# Author: Melyssa Minto

```{r setup}
# Loading packages --------------------------------------------------------
library(tidyverse)
library("ggpmisc")
library(GenomicRanges)
library(rtracklayer)
library(Gviz) 
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(biomaRt)
library(GenomicInteractions)
options(ucscChromosomeNames=FALSE) 
```

# Reading in data ---------------------------------------------------------
```{r setup}

# > Frank et al data ------------------------------------------------------
# source: https://www.nature.com/articles/nn.3995
# peaks were called with MACS v1.4.2 
# "raw read counts mapped to the union of Zic ChIP-seq peak calls for all P7 and P60 replicates were used as input to DESeq"

p7_1_old <- read_delim("../../sequencing_data/Frank_2015/GSM1486431_CbP7_Zic_ChIPseq_peaks_1.bed.gz", 
                        delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
p7_2_old <- read_delim("../../sequencing_data/Frank_2015/GSM1486432_CbP7_Zic_ChIPseq_peaks_2.bed.gz", 
                       delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
p60_1_old <- read_delim("../../sequencing_data/Frank_2015/GSM1486433_CbP60_Zic_ChIPseq_peaks_1.bed.gz", 
                       delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
p60_2_old <- read_delim("../../sequencing_data/Frank_2015/GSM1486434_CbP60_Zic_ChIPseq_peaks_2.bed.gz", 
                       delim = "\t", escape_double = FALSE,  col_names = FALSE, trim_ws = TRUE)

# > Reprocessed  ----------------------------------------------------------
# peaks were called with MACS v2.1.2
p7_1 <- read_delim("../../sequencing_data/preprocessed_data/zic_chip/SRR1557091/macs_n.fdr01_peaks.narrowPeak", 
                   delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
p7_2 <- read_delim("../../sequencing_data/preprocessed_data/zic_chip/SRR1557092/macs_n.fdr01_peaks.narrowPeak", 
                   delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
p60_1 <- read_delim("../../sequencing_data/preprocessed_data/zic_chip/SRR1557093/macs_n.fdr01_peaks.narrowPeak", 
                   delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)
p60_2 <- read_delim("../../sequencing_data/preprocessed_data/zic_chip/SRR1557094/macs_n.fdr01_peaks.narrowPeak", 
                   delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

# merged peaks 
mergedPeaks <- read_delim("../../results/DiffExp_ZicChIP/zic.bed", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

# > Chromatin Links   ------------------------------------------------------
# Source: Yamada et al. 2019 https://www.nature.com/articles/s41586-019-1190-7
plac_file <- read_delim("../../sequencing_data/Yamada/combined_MAPS_peaks.txt", 
                         delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)



# > Peaksets   -------------------------------------------------------------
DNaseP7 <- read_delim("../../results/mergedPeaks/DNaseP7.bed", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

DNaseP60 <- read_delim("../../results/mergedPeaks/DNaseP60.bed", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

H3K27acP7 <- read_delim("../../results/mergedPeaks/H3K27acP7.bed", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

H3K27acP60 <- read_delim("../../results/mergedPeaks/H3K27acP60.bed", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

ZicP7 <- read_delim("../../results/mergedPeaks/ZicP7.bed", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)

ZicP60 <- read_delim("../../results/mergedPeaks/ZicP60.bed", 
    delim = "\t", escape_double = FALSE, col_names = FALSE, trim_ws = TRUE)


```

# Comparing old v new peaks -----------------------------------------------

```{r}
# wraning data

peakData = data.frame( stage = c(rep("p7", 4), rep("p60", 4)),
            replicate = rep(c(1,2), 4),
            version = rep(c("new", "new","old", "old"), 2),
            nPeaks = c(nrow(p7_1),nrow(p7_2),  nrow(p7_1_old), nrow(p7_2_old), nrow(p60_1), nrow(p60_2),nrow(p60_1_old),nrow(p60_2_old)) )

window = 10000
cmd1 = paste0("bedtools window -a ../../sequencing_data/Frank_2015/GSM1486431_CbP7_Zic_ChIPseq_peaks_1.bed.gz  -b ../../sequencing_data/preprocessed_data/SRR1557091/macs_n.fdr01_peaks.narrowPeak -w ", window, "  | wc -l")
cmd2 = paste0("bedtools window -a ../../sequencing_data/Frank_2015/GSM1486432_CbP7_Zic_ChIPseq_peaks_2.bed.gz  -b ../../sequencing_data/preprocessed_data/SRR1557092/macs_n.fdr01_peaks.narrowPeak -w ", window, " | wc -l")
cmd3 = paste0("bedtools window -a ../../sequencing_data/Frank_2015/GSM1486433_CbP60_Zic_ChIPseq_peaks_1.bed.gz  -b ../../sequencing_data/preprocessed_data/SRR1557093/macs_n.fdr01_peaks.narrowPeak -w ", window, " | wc -l")
cmd4 = paste0("bedtools window -a ../../sequencing_data/Frank_2015/GSM1486434_CbP60_Zic_ChIPseq_peaks_2.bed.gz  -b ../../sequencing_data/preprocessed_data/SRR1557094/macs_n.fdr01_peaks.narrowPeak -w ", window, " | wc -l")

overlap_df = data.frame(sample = c("p7 1", "p7 2", "p60 1", "p60 2"),
           overlap = c(system( cmd1, intern = TRUE), system( cmd2, intern = TRUE), system( cmd3, intern = TRUE), system( cmd4, intern = TRUE))
)

overlap_df
```

```{r, fig.height=3, fig,width = 2}
# > number of peaks -------------------------------------------------------
peakData %>% 
  mutate(sample = paste(stage, replicate)) %>% 
  mutate(label = paste0(stage, replicate, version)) %>% 
  left_join(overlap_df) %>% 
  ggplot(aes(x = version, y = nPeaks, fill = version )) +
  geom_col() +
  # geom_text( aes(x = 1.5, y = 55000, label = paste0("nOverlap = ", overlap)))+
  facet_grid(~sample, scales = "free_x")+
  theme_classic() +
  theme(axis.text.x = element_text( angle = 45, hjust = 1)) +
  labs(y = "number of peaks", x = "")
ggsave("../../figures/numpeaks.png")

```
```{r, fig.height=3, fig.width=11}



# import bigwigs
p7_bw <- import.bw("../../sequencing_data/preprocessed_data/SRR1557091/SRR1557091_norm.bw", as="GRanges") 
p7_2bw <- import.bw("../../sequencing_data/preprocessed_data/SRR1557092/SRR1557092_norm.bw", as="GRanges") 

p60_bw <- import.bw("../../sequencing_data/preprocessed_data/SRR1557093/SRR1557093_norm.bw", as="GRanges") 
p60_2bw <- import.bw("../../sequencing_data/preprocessed_data/SRR1557094/SRR1557094_norm.bw", as="GRanges") 


p7_gr = GRanges(seqnames = p7_1$X1, ranges = IRanges(p7_1$X2, p7_1$X3))
p7_2gr = GRanges(seqnames = p7_2$X1, ranges = IRanges(p7_2$X2, p7_2$X3))

p60_gr = GRanges(seqnames = p60_1$X1, ranges = IRanges(p60_1$X2, p60_1$X3))
p60_2gr = GRanges(seqnames = p60_2$X1, ranges = IRanges(p60_2$X2, p60_2$X3))

merged_gr = GRanges(seqnames = mergedPeaks$X1, ranges = IRanges(mergedPeaks$X2, mergedPeaks$X3))
bm <- useMart(biomart = "ENSEMBL_MART_ENSEMBL", 
              dataset = "mmusculus_gene_ensembl")

# plot elements
customFromTxDb <- GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene) 

genomeAxis <- GenomeAxisTrack(name="MyAxis") 
p7_dt <- DataTrack(p7_bw, genome = "mm10",name = "P7") 
p7_2dt <- DataTrack(p7_2bw, genome = "mm10",name = "P7") 

p60_dt <- DataTrack(p60_bw, genome = "mm10", name = "P60") 
p60_2dt <- DataTrack(p60_2bw, genome = "mm10", name = "P60") 

p7_peaks = AnnotationTrack(p7_gr, genome = "mm10", name = "P7 peaks", fill = "blue")
p7_2peaks = AnnotationTrack(p7_2gr, genome = "mm10", name = "P7 peaks", fill = "blue")

p60_peaks = AnnotationTrack(p60_gr, genome = "mm10", name = "P60 peaks", fill = "red")
p60_2peaks = AnnotationTrack(p60_2gr, genome = "mm10", name = "P60 peaks", fill ="red")

merged_peaks = AnnotationTrack(merged_gr, genome = "mm10", name = "merged peaks", fill ="purple")

# plots

png("../../figures/cbln3.png", units = "in", res = 300, height = 2, width = 6)
plotTracks(list(merged_peaks,
                p7_peaks,
                p7_2peaks,
                p60_peaks,
                p60_2peaks,
                p7_dt,
                p7_2dt,
                p60_dt,
                p60_2dt,
                customFromTxDb,
                genomeAxis),  
           from=55878913,to=55896781, 
           chromosome="chr14",
           type="hist",
           window = 5000,
           ylim = c(0, 3),
           transcriptAnnotation = "symbol")

dev.off()

png("../../figures/elavl2.png", units = "in", res = 300, height = 2, width = 6)
plotTracks(list(merged_peaks,
                p7_peaks,
                p7_2peaks,
                p60_peaks,
                p60_2peaks,
                p7_dt,
                p7_2dt,
                p60_dt,
                p60_2dt,
                customFromTxDb,
                genomeAxis),  
           from=91250763,to=91330000, 
           chromosome="chr4",
           type="hist",
           window = 5000,
           ylim = c(0, 2),
           transcriptAnnotation = "symbol")

dev.off()



png("../../figures/grin2c.png", units = "in", res = 300, height = 2, width = 6)
plotTracks(list(merged_peaks,
                p7_peaks,
                p7_2peaks,
                p60_peaks,
                p60_2peaks,
                p7_dt,
                p7_2dt,
                p60_dt,
                p60_2dt,
                customFromTxDb,
                genomeAxis),  
           from=115249169,to=115267243, 
           chromosome="chr11",
           type="hist",
           window = 1000,
           ylim = c(0, 3),
           transcriptAnnotation = "symbol")

dev.off()



```

```{r, fig.height=6, fig.width=11}
options(ucscChromosomeNames=FALSE)
# prepare data
p7_bw <- import.bw("../../sequencing_data/preprocessed_data/zic_chip/SRR1557091/SRR1557091_norm.bw", as="GRanges") 
p7_2bw <- import.bw("../../sequencing_data/preprocessed_data/zic_chip/SRR1557092/SRR1557092_norm.bw", as="GRanges") 

p60_bw <- import.bw("../../sequencing_data/preprocessed_data/zic_chip/SRR1557093/SRR1557093_norm.bw", as="GRanges") 
p60_2bw <- import.bw("../../sequencing_data/preprocessed_data/zic_chip/SRR1557094/SRR1557094_norm.bw", as="GRanges") 


#> zic
p7_zic_gr = GRanges(seqnames = ZicP7$X1, ranges = IRanges(ZicP7$X2, ZicP7$X3))
p60_zic_gr = GRanges(seqnames = ZicP60$X1, ranges = IRanges(ZicP60$X2, ZicP60$X3))

#> dnase
p7_dnase_gr = GRanges(seqnames = DNaseP7$X1, ranges = IRanges(DNaseP7$X2, DNaseP7$X3))
p60_dnase_gr = GRanges(seqnames = DNaseP60$X1, ranges = IRanges(DNaseP60$X2, DNaseP60$X3))

#> k27ac
p7_h3k27ac_gr = GRanges(seqnames = H3K27acP7$X1, ranges = IRanges(H3K27acP7$X2, H3K27acP7$X3))
p60_h3k27ac_gr = GRanges(seqnames = H3K27acP60$X1, ranges = IRanges(H3K27acP60$X2, H3K27acP60$X3))

#>promoter links
plac_gr = GenomicInteractions(GRanges(seqnames = plac_file$X1, ranges = IRanges(plac_file$X2, plac_file$X3)), 
                              GRanges(seqnames = plac_file$X4, ranges = IRanges(plac_file$X5, plac_file$X6)))


# make plot items
customFromTxDb <- GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene) 

genomeAxis <- GenomeAxisTrack(name="MyAxis") 
p7_dt <- DataTrack(p7_bw, genome = "mm10",name = "Zic P7") 
p7_2dt <- DataTrack(p7_2bw, genome = "mm10",name = "Zic P7") 

p60_dt <- DataTrack(p60_bw, genome = "mm10", name = "Zic P60") 
p60_2dt <- DataTrack(p60_2bw, genome = "mm10", name = "Zic P60") 

p7_zic_track = AnnotationTrack(p7_zic_gr, genome = "mm10", name = "P7 Zic", fill = "blue")
p60_zic_track = AnnotationTrack(p60_zic_gr, genome = "mm10", name = "P60 Zic", fill = "red")

p7_dnase_track = AnnotationTrack(p7_dnase_gr, genome = "mm10", name = "P7 Dnase", fill = "blue")
p60_dnase_track = AnnotationTrack(p60_dnase_gr, genome = "mm10", name = "P60 Dnase", fill ="red", , chromosome = "chr11")

p7_h3k27ac_track = AnnotationTrack(p7_h3k27ac_gr, genome = "mm10", name = "P7 H3K27ac", fill = "blue")
p60_h3k27ac_track = AnnotationTrack(p60_h3k27ac_gr, genome = "mm10", name = "P60 H3k27ac", fill ="red")

interaction_track <- InteractionTrack(plac_gr, name = "H3K4me3 PLAC-seq", fill="green")


# plot
plotTracks(list(interaction_track,
                p7_zic_track,
                p60_zic_track,
                p7_dnase_track,
                #p60_dnase_track,
                p7_h3k27ac_track,
                p60_h3k27ac_track,
                p7_dt,
                p7_2dt,
                p60_dt,
                p60_2dt,
                customFromTxDb,
                genomeAxis),  
           from=115210000,to=115268500, 
           chromosome="chr11",
           type="hist",
           window = 1000,
           ylim = c(0, 3),
           sizes = c(1,1,1,1,1,1,1,1,1,1,1,1), 
           transcriptAnnotation = "symbol",
           rotation.title = 1)

```


