---
title: "New figure 1"
author: "Melyssa Minto"
date: "4/29/2022"
output: html_document
---

```{r}
library(png)
library(tidyverse)
library(cowplot)
library(gridExtra)
library(ChIPseeker)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(cowplot)
library(ggmosaic)
library(pheatmap)
library(ggseqlogo)
library(BSgenome.Mmusculus.UCSC.mm10)
library(ggforce)

 # set scheme
scheme <- getScheme(name=getOption("Gviz.scheme"))
scheme$GdObject$col.axis = "white"
scheme$GdObject$background.title = "black"
scheme$GdObject$fontcolor = "black"
scheme$GdObject$rotation.title = 90
scheme$GeneRegionTrack$transcriptAnnotation = "symbol"
scheme$GeneRegionTrack$col = "black"
scheme$GeneRegionTrack$fill = "black"
scheme$GeneRegionTrack$collapseTranscripts = FALSE
scheme$DataTrack$fill.histogram = "darkblue"
scheme$DataTrack$col.histogram = "darkblue"
scheme$GdObject$col.border.title = "transparent"
scheme$HighlightTrack$col = "grey"
scheme$HighlightTrack$fill = "#e3e8ff"
addScheme(scheme, "myScheme")
options(Gviz.scheme="myScheme")
options(ucscChromosomeNames=FALSE)

```
```{r}

plot_track <- function(tracks, from, to, chr,highlight, w= 1000,s, expand_l= 0, expand_r = 0, ylim = c(0,3)){
  
  # setting up plot inputs
 
  # > gene track
  start = from - expand_l
  end = to + expand_r  
  cat(chr, ":", start, "-", end)
  
  gencode <- UcscTrack(genome = "mm10", chromosome = chr,
                      track = "All GENCODE VM25", from = start, to = end,
                      trackType = "GeneRegionTrack",
                      rstarts = "exonStarts", rends = "exonEnds",
                      gene = "name",  symbol = "name2",
                      transcript = "name", strand = "strand",
                      name = "transcripts", showID=TRUE)
  
  # > highlight track
  ht1 <- HighlightTrack(trackList = tracks, start = highlight, width = w, chromosome = chr, )
  


  ts = list(ht1, gencode, genomeAxis)
    #plot
    plotTracks(ts,  
               from=start,
               to=end, 
               chromosome=chr,
               type="hist",
               window = 1000,
               ylim = ylim,
               sizes = s)

  
  
  
}
```

```{r}
zic_peak_data <- read_delim("../results/FinalTables/zic_peak_data.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE)
gene_expr = read_tsv("../results/invivo/DiffExp_RNA/normalized_counts.tsv")
zic_anno_dat =  read_tsv("../results/FinalTables/zic_annotations.txt")

P60vP7_zic = read_tsv("../results/invivo/DiffExp_ZicChIP/ZicChIPDA_data.tsv")

```



```{r zicMA}
nums = table(P60vP7_zic$zic_sig)
zic_ma = P60vP7_zic %>% 
  dplyr::filter(zic_sig != "filtered") %>% 
  ggplot(aes(x = log10(baseMean), y = log2FoldChange, color = zic_sig))+
  geom_point(size = 0.5, alpha = 0.5) +
  scale_color_manual(values= c("blue", "black", "red")) +
  geom_text(x = 3.5, y = 5, label = paste0("Increased Zic Binding\n n = ",nums["UP"] ), color = "red")+
  geom_text(x = 3.5, y = -6, label = paste0("Decreased Zic Binding\n n = ",nums["DOWN"] ), color = "blue")+
  theme_classic() + 
  xlim(c(.7, 4.7))+
  labs( x = "log10(Avg Zic ChIP Signal)", y = "P60 / P7 Log2FC")+
  theme(legend.position = "none") 

zic_ma
```

```{r reads_by_change}

# There are 108 peaks whose mean is greater than 1000 
p7_mean_1000 = P60vP7_zic %>% 
  dplyr::filter(p7_mean > 1000) %>% 
  pull(PeakID)

p60_mean_1000 = P60vP7_zic %>% 
  dplyr::filter(p7_mean > 1000) %>% 
  pull(PeakID)

peaks_mean_1000 = union(p7_mean_1000, p60_mean_1000)


# Distribution of mean signal for early and late peaks at P7 and P60
DistReads=  P60vP7_zic %>% 
  dplyr::filter(!(zic_sig %in% c("filtered", "N.S.")) )%>% 
  pivot_longer(cols = ends_with("_mean"), values_to = "means", names_to = "time") %>% 
  dplyr::filter(means <= 500) %>% 
  mutate(time = factor(time, levels = c("p7_mean", "p60_mean"))) %>%
  ggplot(aes(x = means)) +
  geom_histogram(fill = "white", color = "black", bins = 50) +
  theme_classic() +
  facet_grid(time~zic_sig, labeller = labeller(zic_sig = c("DOWN" = "Early", "UP" = "Late", "N.S." = "Static"), time = c("p60_mean" = "P60", "p7_mean" = "P7")))+
  labs(x = "Mean Zic ChIP Normalized Reads", y = "Num. Peaks")
DistReads
```

```{r loss-gain}
low_exp = 10
lossGain = P60vP7_zic %>% 
  dplyr::filter(p7_mean <= low_exp | p60_mean <= low_exp) %>% 
  dplyr::filter(zic_sig %in% c("UP", "DOWN")) %>% 
  ggplot(aes(x = zic_sig), width = 5)+
  geom_bar(fill = "black", color = "black") +
  theme_classic() +
  scale_x_discrete(limits = c("DOWN", "UP"), labels = c("Early", "Late"))+
  labs(y = "Num. Peaks Mean Signal < 10", x ="Differential Zic Region") 
lossGain

P60vP7_zic %>% 
  dplyr::filter(p7_mean <= low_exp | p60_mean <= low_exp) %>% 
  dplyr::filter(zic_sig %in% c("UP", "DOWN")) %>%  
   dplyr::select(Chr, Start, End) %>%
   write_tsv("../results/invivo/DiffExp_ZicChIP/complete_gain_loss.bed", col_names = F)
 
P60vP7_zic %>% 
  dplyr::filter(p7_mean <= low_exp | p60_mean <= low_exp) %>% 
  dplyr::filter(zic_sig %in% c("UP", "DOWN")) %>% 
  dplyr::count(zic_sig)



```

```{r loss-gain ex, fig.height=6, fig.width=15}
# formatting data
genomeAxis <- GenomeAxisTrack(name="MyAxis") 
customFromTxDb <- GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene)
# make bigwig tracks
p7_zic_bw1 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557091/SRR1557091_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
p7_zic_bw2 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557092/SRR1557092_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
p60_zic_bw1 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557093/SRR1557093_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP60", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
p60_zic_bw2 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557094/SRR1557094_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP60", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 

tracks = list(p7_zic_bw1, 
              p7_zic_bw2, 
              p60_zic_bw1, 
              p60_zic_bw2)
# examples:
  
  




png("../figures/loss_gain_1.png", units = "in", height = 3.5, width = 12, res = 300)
plot_track(tracks,from = 73734300, to = 73757154,chr = "chr18" , highlight = c(73738700),w = 800, s = c(1,1,1,1,.5,1)) ##
dev.off()

png("../figures/loss_gain_2.png", units = "in", height = 3.5, width = 12, res = 300)
plot_track(tracks,from = 72868074,to = 72940073,chr = "chr18", highlight = c(72871933, 72887307, 72929318, 72931785 ),s = c(1,1,1,1,.5,1) ) ##
dev.off()

lg1 = rasterGrob(readPNG("../figures/loss_gain_1.png"))
lg2 = rasterGrob(readPNG("../figures/loss_gain_2.png"))

lg_examples = plot_grid(lg1, lg2, ncol = 1)

```

```{r logos, fig.height= 5, fig.width=3}
seqs= zic_peak_data %>% 
  dplyr::select(sequence_name, zic_sig, motif_id, matched_sequence, zic_Chr, zic_Start, zic_End) %>% 
  distinct() %>% 
  dplyr::mutate( width = (zic_End - zic_Start), 
                 mid = zic_Start + (width/2),
                 left = mid - 10, 
                 right = mid + 10) %>% 
  drop_na(matched_sequence) %>% 
  dplyr::mutate(mid_peak_seq = as.data.frame(Biostrings::getSeq(BSgenome.Mmusculus.UCSC.mm10, zic_Chr, left, right))$x) 




# plotting matched sequences
seqs_down = seqs %>% 
  dplyr::filter(zic_sig == "DOWN") %>%
  pull(matched_sequence)

seqs_up = seqs %>% 
  dplyr::filter(zic_sig == "UP") %>%
  pull(matched_sequence)

seqs_ns = seqs %>% 
  dplyr::filter(zic_sig == "N.S.") %>%
  pull(matched_sequence)


logo_down = ggseqlogo(Biostrings::consensusMatrix(seqs_down),  method = 'bits', seq_type='dna')
logo_up = ggseqlogo(Biostrings::consensusMatrix(seqs_up),  method = 'bits', seq_type='dna')
logo_ns = ggseqlogo(Biostrings::consensusMatrix(seqs_ns),  method = 'bits', seq_type='dna')

plot_grid(logo_down, logo_ns, logo_up, nrow = 1, labels = c("Early", "Static", "Late"), hjust = -4)

# plotting matched sequences by Zic1/2
seqs_down_1 = seqs %>% 
  dplyr::filter(zic_sig == "DOWN", motif_id == "ZIC1_MOUSE.H11MO.0.B") %>%
  pull(matched_sequence)

seqs_down_2 = seqs %>% 
  dplyr::filter(zic_sig == "DOWN", motif_id == "ZIC2_MOUSE.H11MO.0.C") %>%
  pull(matched_sequence)

seqs_up_1 = seqs %>% 
  dplyr::filter(zic_sig == "UP", motif_id == "ZIC1_MOUSE.H11MO.0.B") %>%
  pull(matched_sequence)

seqs_up_2 = seqs %>% 
  dplyr::filter(zic_sig == "UP", motif_id == "ZIC2_MOUSE.H11MO.0.C") %>%
  pull(matched_sequence)

seqs_ns_1 = seqs %>% 
  dplyr::filter(zic_sig == "N.S.", motif_id == "ZIC1_MOUSE.H11MO.0.B") %>%
  pull(matched_sequence)
seqs_ns_2 = seqs %>% 
  dplyr::filter(zic_sig == "N.S.", motif_id == "ZIC2_MOUSE.H11MO.0.C") %>%
  pull(matched_sequence)


logo_down1 = ggseqlogo(Biostrings::consensusMatrix(seqs_down_1),  method = 'bits', seq_type='dna')
logo_up1 = ggseqlogo(Biostrings::consensusMatrix(seqs_up_1),  method = 'bits', seq_type='dna')
logo_ns1 = ggseqlogo(Biostrings::consensusMatrix(seqs_ns_1),  method = 'bits', seq_type='dna')
logo_down2 = ggseqlogo(Biostrings::consensusMatrix(seqs_down_2),  method = 'bits', seq_type='dna')
logo_up2 = ggseqlogo(Biostrings::consensusMatrix(seqs_up_2),  method = 'bits', seq_type='dna')
logo_ns2 = ggseqlogo(Biostrings::consensusMatrix(seqs_ns_2),  method = 'bits', seq_type='dna')


plot_grid(
  plot_grid(logo_down1, logo_ns1, logo_up1, nrow = 1, labels = c("Early", "Static", "Late"), hjust = -1.5),
  plot_grid(logo_down2, logo_ns2, logo_up2, nrow = 1),
  labels = c("Zic1", "Zic2"),
  ncol = 1
)


# plotting middle of sequence
seqs_down = seqs %>% 
  dplyr::select(sequence_name, zic_sig, mid_peak_seq) %>% 
  distinct() %>% 
  dplyr::filter(zic_sig == "DOWN") %>%
  pull(mid_peak_seq)

seqs_up = seqs %>% 
  dplyr::select(sequence_name, zic_sig, mid_peak_seq) %>% 
  distinct() %>% 
  dplyr::filter(zic_sig == "UP") %>%
  pull(mid_peak_seq)

seqs_ns = seqs %>% 
  dplyr::select(sequence_name, zic_sig, mid_peak_seq) %>% 
  distinct() %>% 
  dplyr::filter(zic_sig == "N.S.") %>%
  pull(mid_peak_seq)


logo_down = ggseqlogo(Biostrings::consensusMatrix(seqs_down),  method = 'bits', seq_type='dna')
logo_up = ggseqlogo(Biostrings::consensusMatrix(seqs_up),  method = 'bits', seq_type='dna')
logo_ns = ggseqlogo(Biostrings::consensusMatrix(seqs_ns),  method = 'bits', seq_type='dna')

plot_grid(logo_down, logo_ns, logo_up, nrow = 1, labels = c("Early", "Static", "Late"), hjust = -4)


# plot zic1 and zic2 motif

zic1s = seqs %>% 
   dplyr::select(sequence_name, motif_id, matched_sequence) %>% 
  dplyr::filter( motif_id == "ZIC1_MOUSE.H11MO.0.B") %>%
  distinct() %>% 
  pull(matched_sequence)

zic2s = seqs %>% 
   dplyr::select(sequence_name, motif_id, matched_sequence) %>% 
  dplyr::filter( motif_id == "ZIC2_MOUSE.H11MO.0.C") %>%
  distinct() %>% 
  pull(matched_sequence)

zic1_logo = ggseqlogo(Biostrings::consensusMatrix(zic1s),  method = 'bits', seq_type='dna' ) + labs(x = "position") + theme(axis.text.x = element_text(size = 7)) 
zic2_logo = ggseqlogo(Biostrings::consensusMatrix(zic2s),  method = 'bits', seq_type='dna') + labs(x = "position") + theme(axis.text.x = element_text(size = 7)) 

plot_grid(zic1_logo, zic2_logo, nrow = 2, labels = c("Zic1", "Zic2"), hjust = -2)


```


```{r motif-count, fig.height=5, fig.width=5}
# plot the proportions of Zic1 and Zic2 sites in Early, Late, and Static peaks
# > counting Zic1 and Zic2 separately 
zic_motif2 = zic_peak_data %>% 
  dplyr::filter(!(sequence_name == "no zic peak")) %>% 
  dplyr::select(sequence_name, zic_sig, motif_id,motif_name, n_motifs_in_peak ) %>% 
  distinct() %>% 
  dplyr::filter(!(zic_sig == "filtered")) %>% 
   ggplot(aes(y = zic_sig,  fill = motif_id)) +
  geom_bar( position = "fill") +
  coord_flip()+
  labs( y = "Differential Zic Region", x = "Prop. of Peaks", fill = "Motif") +
  scale_fill_manual(values = c("#143589", "#E1BE6A",  "Black"), limits = c("ZIC1_MOUSE.H11MO.0.B", "ZIC2_MOUSE.H11MO.0.C", "no motif"),labels = c("Zic1", "Zic2", "No Zic1/2 Motif")) +
  scale_y_discrete(limits = c("DOWN", "N.S.", "UP"), labels = c("Early", "Static", "Late"))+
  theme_classic() +
  theme( panel.grid = element_blank())

#> Using Zic1 and Zic2 occurance as a separate category
plt_dat = zic_peak_data %>% 
  dplyr::filter(!(sequence_name == "no zic peak")) %>% 
  dplyr::select(sequence_name, zic_sig, motif_id,motif_name, n_motifs_in_peak ) %>% 
  distinct() %>% 
  dplyr::filter(!(zic_sig == "filtered")) %>% 
  group_by(sequence_name) %>% 
  dplyr::mutate(unique_types = n_distinct(motif_id)) %>% 
  ungroup() %>% 
  dplyr::mutate(motifs = case_when( unique_types == 1 & motif_id == "no motif" ~ "No Zic Motif",
                                    unique_types == 1 & motif_id == "ZIC1_MOUSE.H11MO.0.B" ~ "Zic1 Only",
                                    unique_types == 1 & motif_id == "ZIC2_MOUSE.H11MO.0.C" ~ "Zic2 Only",
                                    unique_types == 2 ~ "Zic1 and Zic2")) 

tab = plt_dat %>% 
  group_by(zic_sig) %>% 
  dplyr::count(motifs) %>% 
  ungroup() %>% 
  pivot_wider(names_from = "zic_sig", values_from = n) %>% 
  column_to_rownames("motifs")

stat_res = bind_rows(
tab %>% 
  dplyr::select(UP, DOWN) %>% 
  prop_test()%>% 
  dplyr::mutate(group1 = "UP", group2 = "DOWN" ),

tab %>% 
  dplyr::select(UP, `N.S.`) %>% 
  prop_test() %>% 
  dplyr::mutate(group1 = "UP", group2 = "N.S." ),

tab %>% 
  dplyr::select(DOWN, `N.S.`) %>% 
  prop_test() %>% 
  dplyr::mutate(group1 = "DOWN", group2 = "N.S." )
) %>% 
  mutate(y.position = c(1.1, 1.03, 1.03))



 zic_motif = plt_dat %>% 
  dplyr::select(sequence_name, zic_sig, motifs) %>% 
  distinct() %>% 
  ggplot(aes(x = zic_sig)) +
  stat_pvalue_manual(stat_res,label = "p = {p}", size = 3.7, tip.length = .25, bracket.shorten = 0.08)+
  geom_bar( aes(fill = motifs), position = "fill") +
  labs( x = "Differential Zic Region", y = "% Peaks", fill = "Motif") +
  scale_fill_manual(values = c("#143589", "#E1BE6A", "#40B0A6", "black"), limits = c("Zic1 Only", "Zic2 Only", "Zic1 and Zic2", "No Zic Motif"), labels = c("Zic1\nOnly", "Zic2\nOnly", "Zic1\nand\nZic2", "No\nZic\nMotif")) +
  scale_x_discrete(limits = c("DOWN","N.S.", "UP"), labels = c("Early", "Static","Late"))+
  scale_y_continuous(labels = scales::percent_format(scale = 100))+
  theme_classic() +
  theme( panel.grid = element_blank(), legend.position = "bottom", legend.key.size = unit(.2, 'cm'))


# Histograms of Zic1 and Zic2 sites in Early, late and Staic peaks

zic_motif_hist = zic_peak_data %>% 
  dplyr::select(zic_PeakID, zic_sig, motif_id, motif_name) %>% 
  distinct() %>% 
  group_by(zic_PeakID, motif_id) %>% 
  add_count(motif_id, name = "n_zics_in_peak") %>% 
  ungroup() %>% 
  mutate(n_zics_in_peak = ifelse(motif_id == "no motif", 0,n_zics_in_peak )) %>% 
  dplyr::select(-motif_name) %>% 
  distinct() %>% 
  dplyr::filter(!(motif_id == "no motif")) %>% 
  ggplot(aes(x = n_zics_in_peak)) + 
  geom_histogram(fill = "white", color = "black", binwidth = 1, boundary=-0.5) +
  scale_x_continuous(breaks=0:5) +
  xlim(0,5)+
  ylim(0, 5000) +
  facet_grid(motif_id ~ zic_sig, labeller = labeller(zic_sig = c("DOWN" = "Early", "UP" = "Late", "N.S." = "Static"), motif_id = c("ZIC1_MOUSE.H11MO.0.B" = "Zic1", "ZIC2_MOUSE.H11MO.0.C" = "Zic2" )))+
  theme_classic()+
  labs(x = "Number of Motifs in Peak", y = "Num. Peaks")

  

zic_motif
zic_motif_hist



```



```{r peak overlap, fig.height=6, fig.width=8}

facet_labs = c("H3K27ac", "DNase", "H3K27ac | DNase" )
names(facet_labs) = c("k27ac_overlap", "dnase_overlap", "k27ac_dnase" )

plt_dat = zic_peak_data %>% 
  dplyr::select(sequence_name, k27ac_sequence_name, dnase_sequence_name, ends_with("sig")) %>% 
  distinct() %>% 
  dplyr::filter(!(sequence_name == "no zic peak")) %>% 
  dplyr::filter(!(zic_sig == "filtered")) %>% 
  dplyr::mutate(k27ac_overlap = ifelse(k27ac_sequence_name !="no overlap" & k27ac_sig %in% c(zic_sig, "N.S.") , 'yes', 'no'),
                dnase_overlap = ifelse(dnase_sequence_name !="no overlap" & dnase_sig  %in% c(zic_sig, "N.S.") , 'yes', 'no'),
                k27ac_dnase = ifelse(k27ac_overlap == "yes" | dnase_overlap == "yes", "yes", "no" )) %>%  
  pivot_longer(cols = c("k27ac_overlap", "dnase_overlap", "k27ac_dnase" )) %>% 
  dplyr::mutate(name = factor(name , levels = names(facet_labs))) 

test_dat = plt_dat %>% 
  dplyr::select(name, value, zic_sig) %>% 
  group_by(zic_sig,name) %>% 
  dplyr::count(value) %>%  
  ungroup() %>% 
  pivot_wider(names_from = zic_sig, values_from = n)

stat_res = bind_rows(
test_dat %>% 
  dplyr::filter(name == "dnase_overlap") %>% 
  dplyr::select(-name) %>% 
  column_to_rownames("value") %>% 
  pairwise_chisq_gof_test() %>% 
  dplyr::mutate(name = "dnase_overlap"),
  
test_dat %>% 
  dplyr::filter(name == "k27ac_overlap") %>% 
  dplyr::select(-name) %>% 
  column_to_rownames("value") %>% 
  pairwise_chisq_gof_test() %>% 
  dplyr::mutate(name = "k27ac_overlap"),

test_dat %>% 
  dplyr::filter(name == "k27ac_dnase") %>% 
  dplyr::select(-name) %>% 
  column_to_rownames("value") %>% 
  pairwise_chisq_gof_test() %>% 
  dplyr::mutate(name = "k27ac_dnase")
) %>%   mutate(y.position = rep(c(45000, 48000, 42000), 3),
               p.adj.text = ifelse(p.adj == 0, "0", format(p.adj, scientific=T, digits = 2)),
               name = factor(name , levels = names(facet_labs)))



zic_overlap = plt_dat %>% 
  ggplot(aes(x = zic_sig))+
  stat_pvalue_manual(stat_res,label = "p = {p.adj.text}", size = 3.7 )+
  geom_bar(aes(fill = value)) +
  scale_x_discrete(limits = c("DOWN", "N.S.", "UP"), labels = c("Early", "Static", "Late")) +
  scale_fill_manual(values = c("grey", "black"))+
  facet_wrap(~name, labeller = labeller(name = facet_labs)) +
  labs( x = "Differential Zic Region", y= "Num. Peaks", fill = "Overlap") +
  theme_classic()+
  theme( panel.grid = element_blank(), legend.position = "bottom")+
  scale_y_continuous(expand = c(0,0), limits = c(0, 50000))



zic_overlap
```





```{r zic loci, fig.height=4, fig.width=6}


zic_loci = zic_anno_dat %>% 
  dplyr::mutate(Feature = fct_relevel(Feature, unique(Feature))) %>% 
  ggplot(aes(x = id, y = Frequency, fill = Feature))+
  geom_col() +
  labs(y = "Percentage(%)", x = "Differential Zic Region") +
  theme_bw() +
  scale_x_discrete( limits =c("P7", "N.S.", "P60"), labels = c("Early", "Static", "Late")) +
  scale_fill_manual(values = c("#bad2e1","#569ea4", "#52af45", "#f88a89", "#f06c45", "#fe870d", "#b294c7","#c7b699", "#b15928"))+
  theme( panel.grid = element_blank()) +
  scale_y_continuous(expand = c(0,0), labels = scales::percent_format(scale = 1)) + 
  coord_flip()

zic_loci2 = zic_anno_dat %>% 
  dplyr::mutate(Feature2 = case_when( Feature %in% "Promoter" ~ "Promoter Proximal",
                                      Feature %in% c("Downstream (<=300)", "Distal Intergenic") ~ "Distal",
                                      TRUE ~ "Gene Body"),
                Feature2 = fct_relevel(Feature2, c("Promoter Proximal", "Gene Body", "Distal"))) %>% 
  ggplot(aes(x = id, y = Frequency, fill = Feature2))+
  geom_col() +
  labs(y = "% Peaks", x = "Differential Zic Region", fill = "Region") +
  theme_classic() +
  scale_x_discrete( limits =c("P7", "N.S.", "P60"), labels = c("Early", "Static", "Late")) +
  scale_fill_manual(values = c("#bad2e1","grey", "#b15928"), labels = c("Promoter\nProximal", "Gene\nBody", "Distal"))+
  theme( panel.grid = element_blank(), legend.position = "bottom", legend.key.size = unit(.2, 'cm')) +
  scale_y_continuous(expand = c(0,0), labels = scales::percent_format(scale = 1))

zic_loci
zic_loci2
```
```{r expr, fig.height=4, fig.width=4}
zic_expr = gene_expr %>% 
  dplyr::filter( SYMBOL %in% c("Zic1", "Zic2")) %>% 
  pivot_longer(cols = starts_with("p")) %>% 
  dplyr::mutate(time = str_extract(name, "[^\\_]+")) %>% 
  ggplot(aes(x = time, y = value)) +
  #scale_fill_manual(values = c("blue", "orange", "red"),limits = c("p7","p14", "p60") )+
  scale_x_discrete( labels = c("P7", "P14", "P60"), limits = c("p7","p14", "p60")) +
  scale_y_continuous(expand = c(0,0), lim = c(0, 50000))+
  stat_summary(fun.data="mean_se",geom="errorbar", color = "black", size = .5, width = .5) +
  stat_summary(fun="mean",geom="bar", fill = "black") +
  facet_wrap(~SYMBOL, ncol = 5) +
  labs(x = "Postnatal Day", y = "DEseq2 \nnormalized counts")+
  theme_classic() +
  theme(
        panel.grid = element_blank(),
         legend.position = "none") 

zic_expr
```




```{r}

P60vP7_zic_dat = zic_peak_data %>% 
  dplyr::select(starts_with("zic")) %>% 
  distinct() %>% 
  dplyr::rename_with( ~ gsub( "zic_","", .x)) 

sa1 = P60vP7_zic_dat %>% 
  dplyr::mutate(size = End - Start) %>% 
  ggplot(aes(x = size) )+ 
  geom_histogram(fill = "white", color = "black", bins = 50) +
  theme_classic() +
  labs(x = "Size of Zic Peaks", y = "Num. Peaks")

P60vP7_zic %>% 
  dplyr::filter(zic_sig != "filtered") %>% 
  dplyr::mutate(size = End - Start) %>% 
  dplyr::summarise(  n = n(),
                     mean = mean(size),
                     sd = sd(size),
                     median = median(size),
                     min = min(size),
                     max = max(size))


sa2 = P60vP7_zic_dat %>% 
  ggplot(aes(x = p7_mean)) +
  geom_histogram(fill = "white", color = "black", bins = 1000) +
  facet_zoom(xlim = c(0, 1000)) +
  theme_classic() +
  labs(x = "Mean signal of Peaks at P7")

sa3 =P60vP7_zic_dat %>% 
  ggplot(aes(x = p60_mean)) +
  geom_histogram(fill = "white", color = "black", bins = 1000) +
  facet_zoom(xlim = c(0, 1000)) +
  theme_classic() +
  labs(x = "Mean signal of Peaks at P60")

ov1 = zic_peak_data %>% 
  dplyr::select(sequence_name, k27ac_sequence_name, dnase_sequence_name, ends_with("sig")) %>% 
  distinct() %>% 
  dplyr::filter(!(sequence_name == "no zic peak")) %>% 
  dplyr::filter(!(zic_sig == "filtered")) %>% 
  dplyr::mutate(k27ac_overlap = ifelse(k27ac_sequence_name !="no overlap" & k27ac_sig %in% c(zic_sig, "N.S.") , 'yes', 'no'),
                dnase_overlap = ifelse(dnase_sequence_name !="no overlap" & dnase_sig  %in% c(zic_sig, "N.S.") , 'yes', 'no')) %>%  
  ggplot(aes(y = dnase_overlap, fill = k27ac_overlap))+
  geom_bar() +
  coord_flip() +
 # facet_wrap(~zic_sig, labeller = labeller(zic_sig = f_labs))+
  scale_fill_manual(values = c("grey", "black"))+
  labs( y = "DNase Overlap", x = "# of Peaks", fill = "K27ac Overlap") +
  theme_classic()+
  theme( panel.grid = element_blank())+
  scale_x_continuous(expand = c(0,0))

ov2= zic_peak_data %>% 
  dplyr::select(sequence_name, k27ac_sequence_name, dnase_sequence_name, ends_with("sig")) %>% 
  distinct() %>% 
  dplyr::filter(!(sequence_name == "no zic peak")) %>% 
  dplyr::filter(!(zic_sig == "filtered")) %>% 
  dplyr::mutate(k27ac_overlap = ifelse(k27ac_sequence_name !="no overlap" & k27ac_sig %in% c(zic_sig, "N.S.") , 'yes', 'no'),
                dnase_overlap = ifelse(dnase_sequence_name !="no overlap" & dnase_sig  %in% c(zic_sig, "N.S.") , 'yes', 'no')) %>%  
  ggplot(aes(y = k27ac_overlap, fill = dnase_overlap))+
  geom_bar() +
  coord_flip() +
 # facet_wrap(~zic_sig, labeller = labeller(zic_sig = f_labs))+
  scale_fill_manual(values = c("grey", "black"))+
  labs( y = "K27ac Overlap", x = "# of Peaks", fill = "DNase Overlap") +
  theme_classic()+
  theme( panel.grid = element_blank())+
  scale_x_continuous(expand = c(0,0))






s1 =plot_grid(sa1, DistReads,zic_motif_hist,  labels = c("A", "B", "C"), nrow = 2)
png("../figures/supp_figure1.png", height = 7, width = 8, units = "in", res = 300)
s1
dev.off()
```


```{r, fig.height=10, fig.width=12}
zic_motif_hist_logo = plot_grid(plot_grid(zic1_logo, zic2_logo, ncol = 1, labels = c("F", "G")),
          zic_motif_hist,
          nrow = 1,
          rel_widths = c(.4, .6),
          labels = c("", "H"),
          hjust = -2)

fig1_horiz = plot_grid(
  plot_grid(plot_grid(zic_ma, sa1, ncol = 1, labels = c("A", "B")), DistReads, lossGain, lg_examples, rel_widths = c(.25, .3,.1 ,.35), nrow = 1, labels = c("","C", "D", "E")),
  plot_grid(zic_motif_hist_logo, zic_motif,zic_overlap, zic_loci2 ,rel_widths = c(.35,.175, .3, .175), nrow = 1, labels = c( "", "I", "J", "K")),
          ncol =1
)
 

fig1_vert = plot_grid(
  plot_grid(zic_ma, DistReads, lossGain, nrow = 1, rel_widths = c(.4, .4, .2), labels = c("A", "B", "C")),
  plot_grid(lg_examples, sa1, zic_motif_hist_logo, nrow = 1, labels = c("D", "E"), vjust = 1, rel_widths = c(.4, .2, .4)),
  plot_grid(zic_motif,zic_overlap, zic_loci2 , nrow = 1, rel_widths = c(.25, .5, .25), labels = c("I", "J", "K")),
          ncol =1
)
 
 
png("../figures/figure1.png", height = 10, width = 12, units = "in", res = 300)
fig1_vert
dev.off()

fig1_vert
```
```{r, fig.height=10, fig.width=16}

plot_grid(
  plot_grid(zic_ma, DistReads, lossGain, nrow = 1, rel_widths = c(.4, .4, .2), labels = c("A", "B", "C")),
  plot_grid(lg_examples, sa1, zic_motif_hist_logo, nrow = 1, labels = c("D", "E"), vjust = 1, rel_widths = c(.4, .2, .4)),
  plot_grid(zic_motif,zic_overlap, zic_loci2 , nrow = 1, rel_widths = c(.25, .5, .25), labels = c("I", "J", "K")),
          ncol =1
)
 

```


