---
title: "Figure 3"
output: html_notebook
---

In this document, we reproduce figures from the Zic mansuscript

```{r setup}
library(tidyverse)
library(readxl)
library(cowplot)
library(ggrepel)
library(ggtext)
library(VennDiagram)
library(clusterProfiler)
library(ReactomePA)
library(AnnotationDbi)
library(org.Mm.eg.db)

 # set scheme
 # set scheme
scheme <- getScheme(name=getOption("Gviz.scheme"))
scheme$GdObject$col.axis = "white"
scheme$GdObject$background.title = "black"
scheme$GdObject$fontcolor = "black"
scheme$GdObject$rotation.title = 90
scheme$GeneRegionTrack$transcriptAnnotation = "symbol"
scheme$GeneRegionTrack$col = "black"
scheme$GeneRegionTrack$fill = "black"
scheme$GeneRegionTrack$collapseTranscripts = FALSE
scheme$DataTrack$fill.histogram = "darkblue"
scheme$DataTrack$col.histogram = "darkblue"
scheme$GdObject$col.border.title = "transparent"
scheme$HighlightTrack$col = "grey"
scheme$HighlightTrack$fill = "#e3e8ff"
addScheme(scheme, "myScheme")
options(Gviz.scheme="myScheme")
options(ucscChromosomeNames=FALSE)
```

```{r define functions}



plot_track <- function(gene, tracks, highlight, w = 1000, expand_l= 0, expand_r = 0, ylim = c(0,3)){
  # getting gene info
  id = mapIds(org.Mm.eg.db, keys =gene , column ="ENTREZID", keytype="SYMBOL")
  info = biomaRt::select(TxDb.Mmusculus.UCSC.mm10.knownGene, keys = id  , columns=c("TXNAME", "TXSTART", "TXEND", "TXCHROM"), keytype="GENEID")
  start = info$TXSTART[1]
  end = info$TXEND[1]
  chr = info$TXCHROM[1]
  
  # setting up plot inputs
  start = start - expand_l
  end = end + expand_r  
  cat(chr, start, end)
  # getting gene track
  gencode <- UcscTrack(genome = "mm10", chromosome = chr,
                      track = "All GENCODE VM25", from = start, to = end,
                      trackType = "GeneRegionTrack", 
                      rstarts = "exonStarts", rends = "exonEnds", 
                      gene = "name",  symbol = "name2", 
                      transcript = "name", strand = "strand",
                      name = "transcripts", showID=TRUE)
 
  
    # > highlight track
  if(is.null(highlight)){
    # create trackset for plotting
    ts = list(tracks, gencode, genomeAxis)
  }else{
    # creating trackset for plotting with highlighting
    ht1 <- HighlightTrack(trackList = tracks, start = highlight, width = w, chromosome = chr)
    ts = list(ht1, gencode, genomeAxis)
  }

    #plot
    # > create a vector of track sizes
    s = c(rep(1, length(tracks) ), 0.5, 1)
    plotTracks(ts,  
               from=start,
               to=end, 
               chromosome=chr,
               type="hist",
               window = 1000,
               ylim = ylim,
               sizes = s)
  
  
}

plot_bart_homer <- function(cond1, bart_data, homer_data, s, title){
  # plot the TFs enriched in BArt and Homer analyses
  dat1 = bart_data %>% 
    # filter out Zics
    dplyr::filter(!grepl("ZIC", TF)) %>% 
    # filter out Zics
    dplyr::filter(gene_sig %in% c("UP", "DOWN")) %>% 
    # filter for factors significantly enriched
    dplyr::filter(bart_sig == "Enriched") %>% 
    # create a label: TF ( Mapped Gene)
    dplyr::mutate(label_f = ifelse(str_to_lower(TF) != str_to_lower(SYMBOL), paste0(TF, "(", SYMBOL, ")"), TF)) 
    
  dat2 = homer_data %>% 
     # filter out Zics
    dplyr::filter(!grepl("ZIC", SYMBOL, ignore.case = T)) %>% 
    # filter out Zics
    dplyr::filter(gene_sig %in% c("UP", "DOWN")) %>% 
    # filter for factors significantly enriched
    dplyr::filter(`q-value (Benjamini)` < 0.05) %>% 
    # create a label: TF ( Mapped Gene)
    dplyr::mutate(label_f = ifelse(str_to_lower(name) != str_to_lower(SYMBOL), paste0(name, "(", SYMBOL, ")"), name))
    
  # get a list of common signifincant TFs between BART and HOMER data sets 
  common_sig_TFs = intersect(str_to_title(dat1$SYMBOL), str_to_title(dat2$SYMBOL))
  # get a list of common TFs between BART and HOMER data sets 
  common_TFs = intersect(str_to_title(bart_data$SYMBOL), str_to_title(homer_data$SYMBOL))

  # wrangling data for plotting
   facet_labels = c("ChIP Enrichment", "Motif Enrichment")
   names(facet_labels) = c("bart", "homer")
  
    bind_rows(list(bart = dat1 %>% 
                                   # addimg * to label if TF is significant in BART and HOPMER
                     dplyr::mutate(label_f = ifelse(str_to_title(SYMBOL) %in% common_sig_TFs, paste0("*", label_f), str_to_title(label_f)),
                                   # making the label blue if TF is in BART and HOPMER databases
                                   label_f = ifelse(str_to_title(SYMBOL) %in% common_TFs, paste0('<span style="color:blue;font-weight:bold">', label_f, '</span>' ), label_f),
                                   # sorting TFs by adjusted p-value
                                   label_f = fct_reorder(as.factor(label_f), get(cond1))) %>% 
                     # renaming column that holds p values for merging 
                     dplyr::rename(enrich_stat = !!cond1) %>% 
                     # select relevant data 
                     dplyr::select(enrich_stat, label_f),
                   homer = dat2 %>% 
                                   # addimg * to label if TF is significant in BART and HOPMER
                     dplyr::mutate(label_f = ifelse(str_to_title(SYMBOL) %in% common_sig_TFs, paste0("*", label_f), str_to_title(label_f)),
                                   # making the label blue if TF is in BART and HOPMER databases
                                   label_f = ifelse(str_to_title(SYMBOL) %in% common_TFs, paste0('<span style="color:blue;font-weight:bold">', label_f, '</span>' ), label_f),
                                   # sorting TFs by adjusted p-value
                                   label_f = fct_reorder(as.factor(label_f), `q-value (Benjamini)`)) %>% 
                     # renaming column that holds p values for merging
                      dplyr::rename(enrich_stat = `q-value (Benjamini)`) %>% 
                     # select relevant data 
                      dplyr::select(enrich_stat, label_f)),
                   .id = "id") %>% 
      ggplot(aes(y = enrich_stat, x = label_f, fill = id)) +
        geom_bar(stat = "identity") +
        geom_hline(yintercept = 0,color = "grey",linetype = "dashed") +
        coord_flip() +
        labs(x = "") +
        facet_grid(id~., space = "free_y", scales = "free_y",  labeller = as_labeller(facet_labels)) +
        #scale_y_continuous(limits = c(-11,5), expand = c(0, 0))+
        scale_fill_manual(limits = c("bart", "homer"), values = c("#FF6308", "black")) +
        theme_classic() +
        theme(axis.line.y = element_blank(),
              axis.text = element_text(size = s),
              axis.text.x = element_markdown(),
              axis.text.y = element_markdown(),
              strip.text = element_text(size = s),
              legend.position = "none",
              plot.title = element_text(hjust = 0.45)) +
        ylab("Adjusted p-value") +
        ggtitle(str_to_title(title))
    


}

plot_distinct <- function(data, cond1, cond2, label1, label2, table_tilte = NULL) {
  # wrangle data for plotting
  dis = data %>% 
    # filter TFs that are DE
    dplyr::filter(gene_sig %in% c("UP", "DOWN")) %>% 
    # Label the distict TFs only
    dplyr::mutate(label = case_when(enriched_in %in% "Similar Enrichment" ~ "" ,
                                    TRUE ~ str_to_title(TF))) %>% 
    # select relative columns
    dplyr::select(-SYMBOL, -gene_sig, -ends_with("mean")) %>% 
    distinct() 
  
  # create a table showing how many distinctly enriched TFs there were
 dis_table = dis$enriched_in %>% 
   fct_relabel(~ str_to_title(gsub("_|_all", " ", .x))) %>% 
   fct_count()
 colnames(dis_table) = c( paste0(table_tilte, "Enrichment"), "n")
 
 # plot the distinct TFs 
 dis %>% 
    ggplot(aes(x = -log10(get(cond1)), y = -log10(get(cond2)), color = enriched_in, label = label )) +
    geom_point(size = 0.5) +
    geom_text_repel(show.legend = F, max.overlaps = 10) +
    theme_classic() +
    labs(x = paste0(gsub("Distinct ", " ", label1), " -log10(adj p-val)"), y = paste0(gsub("Distinct ", " ", label2), " -log10(adj p-val)") ) +
    scale_color_manual(name = "", limits = c(cond1, cond2, "Similar Enrichment"), labels = c(label1, label2, "Similar \nEnrichment"), values = c("dodgerblue4", "darkorange4", "grey") ) +
    annotate(geom = "table", label = list(dis_table), x = 5, y = 5, fill = "white" ,  table.theme = ttheme_gtlight)+
    scale_y_continuous(limits = c(0, 5)) +
    scale_x_continuous(limits = c(0, 5)) +
    theme(legend.position = "bottom") +
    guides(colour = guide_legend(override.aes = list(size=3)))
}

distinct_lolipop  <- function(data, cond1, cond2, gene_expr_level) {
  # plot set up
   facet_labels = c('<span style="color:dodgerblue4">Early</span>', '<span style="color:darkorange4">Late</span>')
   names(facet_labels) = c("Early", "Late")
   
  # extract data
  d1 = data %>%  
    # determine whether each TF is enriched in one set v another set
    dplyr::mutate(enrich_sig = ifelse(enriched_in == cond1, get(cond1), get(cond2))) %>% 
    # filter for significant TFs in set1
    dplyr::filter(enriched_in %in% c(cond1)) %>% 
    # remove significance column
    dplyr::select(-all_of(cond1)) %>% 
    # cleaning up TF names and creating label for plotting
    dplyr::mutate(TF = gsub("-", "", TF)) %>% 
    dplyr::mutate(label = ifelse(str_to_lower(TF) != str_to_lower(SYMBOL), paste0(TF, "(", SYMBOL, ")"), TF),
                  label = fct_reorder(label, -enrich_sig)) 
  
  d2 =  data %>% 
    # determine whether each TF is enriched in one set v another set
    dplyr::mutate(enrich_sig = ifelse(enriched_in == cond1, get(cond1), get(cond2))) %>% 
    # filter for significant TFs in set2
    dplyr::filter(enriched_in %in% c(cond2)) %>% 
     # remove significance column
    dplyr::select( -all_of(cond2)) %>% 
    # cleaning up TF names and creating label for plotting
    dplyr::mutate(TF = gsub("-", "", TF)) %>% 
    dplyr::mutate(label = ifelse(str_to_lower(TF) != str_to_lower(SYMBOL), paste0(TF, "(", SYMBOL, ")"), TF),
                  label = fct_reorder(label, -enrich_sig))
  
  # plotting data
    bind_rows(
      list( Late = d2  %>%
              # filter out ot time specific gene expression
              dplyr::filter(p60_mean > gene_expr_level) %>% 
              # filter for time specific gene regulation
              dplyr::filter(gene_sig %in% c("UP")) %>% 
              # sort TF by log fold change
              dplyr::mutate(label = fct_reorder(label, log2FoldChange)) %>% 
              # renaming gene means for plotting
              dplyr::rename(plt_mean = p60_mean),
            Early = d1 %>% 
              # filter out ot time specific gene expression
              dplyr::filter(p7_mean > gene_expr_level) %>% 
               # filter for time specific gene regulation
              dplyr::filter(gene_sig %in% c( "DOWN")) %>% 
              # sort TF by log fold change
              dplyr::mutate(label = fct_reorder(label, -log2FoldChange))%>% 
              # renaming gene means for plotting
              dplyr::rename(plt_mean = p7_mean)),
      .id = "id") %>% 
    # plot  
    ggplot(aes(y = log2FoldChange , x = label, color = enrich_sig, size = plt_mean )) +
    geom_segment( aes(x=label, xend=label, y=0, yend=log2FoldChange, size = 1), color = "grey", show.legend = FALSE) +
    geom_hline(yintercept = 0, color = "black", size = .5, linetype = "dashed") +
    geom_point() +
    theme_classic() +
    scale_x_discrete(labels = function(x) str_wrap(x, width = 5))+
    scale_color_viridis(limits = c(0, 0.05))+
    facet_grid(id~., scales = "free", space = "free_y", labeller = labeller(facet_labels)) +
    coord_flip() +
    theme(legend.position = "bottom", 
          legend.text = element_text(angle = 45, hjust = 1),
          strip.text.x = element_markdown() , 
          axis.text.y = element_text(size = 8),
          panel.spacing=unit(0,"pt"), 
          plot.title = element_markdown()) +
    labs(x = "Enriched TFs" , y = "LFC", color = "TF Enrichment", size = "expression")   

}

plot_overlap <-function(peakset, overlap, name){
  # prepping data for join
  # > peakset of choice
  peakset = peakset %>% 
    # create a unique peak identifier chr:start-end
    dplyr::mutate(peak_name = paste0(X1, ":", X2,"-", X3)) %>% 
    # select unique id 
    dplyr::select(peak_name) %>% 
    distinct() %>% 
    ungroup()
  
  # > overlaping peaks output from bedtools
  overlap = overlap %>% 
     # create a unique peak identifier for zic peaks and peakset of choice
      dplyr::mutate(zic_name = paste0(X1, ":", X2,"-", X3),
                    peak_name = paste0(X4, ":", X5,"-", X6)) %>% 
    dplyr::select(peak_name, zic_name) %>% 
    distinct() %>% 
    ungroup()
  
  # > Zic peak DE data
  zic_peakset = P60vP7_zic %>% 
    # # create a unique peak identifier chr:start-end
    dplyr::mutate(zic_name = paste0(Chr, ":", Start,"-", End)) %>% 
    # select relevant columns for plotting
    dplyr::select(zic_name, zic_sig) %>% 
    distinct() %>% 
    ungroup()
  
  # join data
  data =  zic_peakset %>% 
    full_join(overlap) %>% 
    full_join(peakset, by = "peak_name") %>% 
    distinct() %>% 
    ungroup()
  

  # plot1: How many peaks intersect with Zic peaks? What proportion of peaks 
  # have Zic co-occupancy
  p1 = data %>% 
    drop_na(peak_name) %>% 
    distinct()  %>% 
                  # create columns for peakset  overlap
    dplyr::mutate(overlap = ifelse(!is.na(zic_name) , "yes", "no"),
                  fill_o = ifelse(overlap == "yes", zic_sig, "no"), 
                  fill_o = fct_relevel(fill_o, c("no","DOWN", "N.S.", "UP")), 
                  peak = paste0(name, " Peak")) %>% 
    # plot
    ggplot(aes(y = peak, fill = fill_o)) +
    geom_bar(position = "fill") +
    coord_flip() +
    scale_fill_manual(limits = c("DOWN", "UP", "N.S.", "no"), labels = c("Early", "Late", "Static", "No \nOverlap"),values = c("blue", "red", "black", "grey")) +
    theme_classic() +
    labs(y = "", x = "Prop. Overlap", fill = "Zic Peak") +
   theme(legend.key.size = unit(.3, 'cm'))

  

  # compare proportion of overlap between Differential Zic Regions
  # statistical test = chi_sq
  results = data %>% 
    drop_na(zic_name) %>% 
    distinct()  %>% 
    dplyr::mutate(overlap = ifelse(!is.na(peak_name) , "yes", "no")) %>% 
    group_by(zic_sig) %>% 
    dplyr::count(overlap) %>% 
    ungroup() %>% 
    dplyr::filter(zic_sig != "filtered") %>% 
    spread(key = zic_sig, value = n) %>% 
    column_to_rownames( var = "overlap") %>% 
    chisq_test()

  
  # plot2: Of the Zic peaks, how many of those are occupied by peakset of 
  # choice
  p2 = data %>% 
    drop_na(zic_name) %>% 
    distinct()  %>% 
    dplyr::mutate(overlap = ifelse(!is.na(peak_name) , "yes", "no")) %>% 
    group_by(zic_sig) %>% 
    dplyr::count(overlap) %>% 
    ungroup() %>% 
    dplyr::filter(zic_sig != "filtered") %>% 
    ggplot(aes(x = zic_sig, y = n, fill = overlap)) +
    geom_bar(position = "fill", stat = "identity") +
    scale_x_discrete(limits = c("DOWN",  "N.S.", "UP"), labels = c("Early", "Static", "Late")) +
    scale_fill_manual(values = c("black", "grey")) +
    theme_classic() +
    #geom_signif(tip_length = 0, annotation = results$p.signif, xmin = 0.8, xmax = 3.2, y_position = 1.01 , color = "black")+
    labs(x = "Differential Zic Regions", y= "Prop. Overlap", fill = paste0("Overlap w/\n", name))  +
   theme(legend.key.size = unit(.3, 'cm'))

  return(list(plot1 = p1, plot2 = p2))
    

}

make_track <- function(peaks, name){
  # make annotation track for overlap of peaks
  AnnotationTrack(GRanges(seqnames = peaks$X1, ranges = IRanges(start = peaks$X2, end = peaks$X3)),
                                  genome = "mm10", name = name, col.title = "white", col.border.title = "black",fill = "black")
}

```

```{r read in data}

# Homer results
activ_homer = read_tsv("../results/invivo/homer_results/activating/homer_results.tsv")
repr_homer = read_tsv("../results/invivo/homer_results/repressive/homer_results.tsv")
early_activ_homer = read_tsv("../results/invivo/homer_results/P7_peaks_DOWNGenes/homer_results.tsv")
early_repr_homer = read_tsv("../results/invivo/homer_results/P7_peaks_UpGenes/homer_results.tsv")
late_activ_homer = read_tsv("../results/invivo/homer_results/P60_peaks_UpGenes/homer_results.tsv")
late_repr_homer = read_tsv("../results/invivo/homer_results/P60_peaks_DOWNGenes/homer_results.tsv")
early_homer = read_tsv("../results/invivo/homer_results/early/homer_results.tsv")
late_homer = read_tsv("../results/invivo/homer_results/late/homer_results.tsv")
early_all_homer = read_tsv("../results/invivo/homer_results/P60vP7_DOWN/homer_results.tsv")
late_all_homer = read_tsv("../results/invivo/homer_results/P60vP7_UP/homer_results.tsv")

# Motif Enrichment Comparisons
earlyVlate_act_hom = read_tsv("../results/invivo/peak_gene/rrho/early_late_activating_homer.txt")
early_actVrep_hom = read_tsv("../results/invivo/peak_gene/rrho/early_activating_repressive_homer.txt")
earlyVlate_rep_hom = read_tsv("../results/invivo/peak_gene/rrho/early_late_repressive_homer.txt")
late_actvrep_hom = read_tsv("../results/invivo/peak_gene/rrho/late_activating_repressive_homer.txt")
actvrep_hom = read_tsv("../results/invivo/peak_gene/rrho/activating_repressive_homer.txt")
earlyvlate_hom = read_tsv("../results/invivo/peak_gene/rrho/early_late_homer.txt")
early_allVloop_hom = read_tsv("../results/invivo/peak_gene/rrho/early_allvloop_homer.txt")
late_allVloop_hom = read_tsv("../results/invivo/peak_gene/rrho/late_allvloop_homer.txt")
earlyvlate_all_hom = read_tsv("../results/invivo/peak_gene/rrho/earlyvlate_all_homer.txt")

# BART resutls
activ_bart = read_tsv("../results/invivo/peak_gene/activating/bart_results.txt")
repr_bart = read_tsv("../results/invivo/peak_gene/repressive/bart_results.txt")
early_activ_bart = read_tsv("../results/invivo/peak_gene/early_activating/bart_results.txt")
early_repr_bart = read_tsv("../results/invivo/peak_gene/early_repressive/bart_results.txt")
late_activ_bart = read_tsv("../results/invivo/peak_gene/late_activating/bart_results.txt")
late_repr_bart = read_tsv("../results/invivo/peak_gene/late_repressive/bart_results.txt")
early_bart = read_tsv("../results/invivo/peak_gene/early/bart_results.txt")
late_bart = read_tsv("../results/invivo/peak_gene/late/bart_results.txt")
early_all_bart = read_tsv("../results/invivo/DiffExp_ZicChIP/early/bart_results.txt")
late_all_bart = read_tsv("../results/invivo/DiffExp_ZicChIP/late/bart_results.txt")

# ChIP Enrichment Comparisons
earlyVlate_act = read_tsv("../results/invivo/peak_gene/rrho/early_late_activating.txt")
early_actVrep = read_tsv("../results/invivo/peak_gene/rrho/early_activating_repressive.txt")
earlyVlate_rep = read_tsv("../results/invivo/peak_gene/rrho/early_late_repressive.txt")
late_actvrep = read_tsv("../results/invivo/peak_gene/rrho/late_activating_repressive.txt")
actvrep = read_tsv("../results/invivo/peak_gene/rrho/activating_repressive.txt")
earlyvlate = read_tsv("../results/invivo/peak_gene/rrho/early_late.txt")
early_allVloop = read_tsv("../results/invivo/peak_gene/rrho/early_allvloop.txt")
late_allVloop = read_tsv("../results/invivo/peak_gene/rrho/late_allvloop.txt")

earlyvlate_all = read_tsv("../results/invivo/peak_gene/rrho/earlyvlate_all.txt")


# ChIP OVerlap
atoh1_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_atoh1.bed", col_names = F) 
atoh_peaks =  read_tsv("../results/invivo/mergedPeaks/atoh1.bed", col_names = F)

ctcfP56_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_ctcfP56.bed", col_names = F) 
ctcfP56_peaks =  read_tsv("../results/invivo/mergedPeaks/ctcfP56.bed", col_names = F)

ctcfP22_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_ctcfP22.bed", col_names = F) 
ctcfP22_peaks =  read_tsv("../results/invivo/mergedPeaks/ctcfP22.bed", col_names = F)

ctcfP4_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_ctcfP4.bed", col_names = F) 
ctcfP4_peaks =  read_tsv("../results/invivo/mergedPeaks/ctcfP4.bed", col_names = F)

rad21P4_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_rad21P4.bed", col_names = F) 
rad21P4_peaks =  read_tsv("../results/invivo/mergedPeaks/rad21P4.bed", col_names = F)

rad21P56_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_rad21P56.bed", col_names = F) 
rad21P56_peaks =  read_tsv("../results/invivo/mergedPeaks/rad21P56.bed", col_names = F)

chd7_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_chd7P4.bed", col_names = F) 
chd7_peaks =  read_tsv("../results/invivo/mergedPeaks/chd7P4.bed", col_names = F)

chd4_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_chd4P22.bed", col_names = F) 
chd4_peaks =  read_tsv("../results/invivo/mergedPeaks/chd4P22.bed", col_names = F)

smc1_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_smc1p22.bed", col_names = F) 
smc1_peaks =  read_tsv("../results/invivo/mergedPeaks/smc1p22.bed", col_names = F)

h2a_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_h2ap22.bed", col_names = F) 
h2a_peaks =  read_tsv("../results/invivo/mergedPeaks/h2ap22.bed", col_names = F)

h3_zic_overlap = read_tsv("../results/invivo/mergedPeaks/zic_h3p22.bed", col_names = F) 
h3_peaks =  read_tsv("../results/invivo/mergedPeaks/h3p22.bed", col_names = F)

#P60vP7_zic_overlap = read_tsv("../results/invivo/peak_gene/zic_atoh_overlap.txt")


# Differential Zic Data
P60vP7_zic = read_tsv("../results/invivo/DiffExp_ZicChIP/ZicChIPDA_data.tsv")

```

# Figure panels

```{r schematic}

a = rasterGrob(readPNG("../figures/rrho_workflow.png"), interpolate = T)

```
![](../figures/rrho_workflow.png)
```{r distinct, fig.height=4, fig.width=8}
# Early v Late
c1 = plot_distinct(earlyvlate, "early", "late", "Distinct Early Looped", "Distinct Late Looped" , table_tilte = "Bart\n")
d1 = plot_distinct(earlyvlate_all, "early_all", "late_all", "Distinct Early", "Distinct Late", table_tilte = "Bart\n" )
ch1 = plot_distinct(earlyvlate_hom, "early", "late", "Distinct Early Looped", "Distinct Late Looped", table_tilte = "Homer\n"  )
dh1 = plot_distinct(earlyvlate_all_hom, "early_all", "late_all", "Distinct Early", "Distinct Late" , table_tilte = "Homer\n" )



plot_grid(c1, ch1, labels = c("Bart", "Homer"), hjust = -2)
plot_grid(d1, dh1, labels = c("Bart", "Homer"), hjust = -2)
```




```{r distinct lolipop, fig.height=6, fig.width=8}
# using the 25-pecentile as a cut off for low reads
gene_expr_filt = quantile(earlyvlate_all$p7_mean, na.rm=T)[2]

chip = distinct_lolipop(earlyvlate, "early", "late", gene_expr_filt )
motif = distinct_lolipop(earlyvlate_hom, "early", "late", gene_expr_filt )
legend_plt = get_legend(motif)
d_tf = plot_grid(plot_grid(motif + theme(legend.position = "none"), chip + theme(legend.position = "none"), nrow = 1, labels = c("E", "F")),
          legend_plt,
          rel_heights = c(.9, .1),
          ncol = 1)

d_tf


chip_all = distinct_lolipop(earlyvlate_all, "early_all", "late_all", gene_expr_filt )
motif_all = distinct_lolipop(earlyvlate_all_hom, "early_all", "late_all", gene_expr_filt )
legend_plt_all = get_legend(motif_all)
d_tf_all = plot_grid(plot_grid(motif_all + theme(legend.position = "none"), chip_all + theme(legend.position = "none"), nrow = 1, labels = c("C", "D")),
          legend_plt_all,
          rel_heights = c(.9, .1),
          ncol = 1)

d_tf_all
```
# Examples:

```{r chip overlap, fig.height=6, fig.width=10}


atoh_p = plot_overlap(atoh_peaks, atoh1_zic_overlap, "P5 - Atoh1")
chd7_p = plot_overlap(chd7_peaks, chd7_zic_overlap, "P4 - Chd7")
chd4_p = plot_overlap(chd4_peaks, chd4_zic_overlap, "P22 - Chd4")
ctcf22_p = plot_overlap(ctcfP22_peaks, ctcfP22_zic_overlap, "P22 - CTCF")
ctcf4_p = plot_overlap(ctcfP4_peaks, ctcfP4_zic_overlap, "P4 - CTCF")
ctcf56_p = plot_overlap(ctcfP56_peaks, ctcfP56_zic_overlap, "P56 - CTCF")
rad21_4_p = plot_overlap(rad21P4_peaks, rad21P4_zic_overlap, "P4 - Rad21")
rad21_56_p = plot_overlap(rad21P56_peaks, rad21P56_zic_overlap, "P56 - Rad21")
smc1_p = plot_overlap(smc1_peaks, smc1_zic_overlap, "P22 - Smc1")
h2a_p = plot_overlap(h2a_peaks, h2a_zic_overlap, "P22 - H2A.z")
h3_p = plot_overlap(h3_peaks, h3_zic_overlap, "P22 - H3")



chip_o = plot_grid(atoh_p$plot1, atoh_p$plot2, 
          chd7_p$plot1, chd7_p$plot2, 
          chd4_p$plot1, chd4_p$plot2,
          rad21_4_p$plot1, rad21_4_p$plot2,
          rad21_56_p$plot1, rad21_56_p$plot2, 
          smc1_p$plot1,  smc1_p$plot2,
          h2a_p$plot1, h2a_p$plot2, 
          nrow =4, 
          rel_widths =  rep(c(.75, 1), 7) )



# png("../figures/supp_figure_chip_overlap.png", height = 6, width = 8, res = 300, units = "in" )
chip_o
# dev.off()
```

```{r tracks, fig.width=12, fig.height=12}

# formatting data
genomeAxis <- GenomeAxisTrack(name="MyAxis") 
customFromTxDb <- GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene)

# make bigwig tracks
p7_zic_bw1 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557091/SRR1557091_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
p7_zic_bw2 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557092/SRR1557092_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
p60_zic_bw1 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557093/SRR1557093_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP60", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
p60_zic_bw2 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557094/SRR1557094_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP60", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
atoh1_bw1 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/atoh1_chip/SRR059286/SRR059286_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Atoh1\n P5", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
atoh1_bw2 <- DataTrack(import.bw("../sequencing_data/preprocessed_data/atoh1_chip/SRR059287/SRR059287_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Atoh1 \nP5", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# chd7_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/chd7_p4_chip/SRR13371039/SRR13371039_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "Chd7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# chd4_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/chd4_p22_chip/SRR3659055/SRR3659055_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "Chd4", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# ctcf_p4_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/ctcf_p4_chip/SRR13371014/SRR13371014_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "CTCF P4", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# ctcf_p22_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/ctcf_p22_chip/SRR10275430/SRR10275430_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "CTCF P22", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# ctcf_p56_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/ctcf_p56_chip/SRR8696015/SRR8696015_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "CTCF P56", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# rad21_p4_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/rad21_p4_chip/SRR13371020/SRR13371020_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "Rad21 P4", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# rad21_p56_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/rad21_p56_chip/SRR8696023/SRR8696023_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "Rad21 P56", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# smc1_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/smc1_p22_chip/SRR10275434/SRR10275434_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "Smc1 P22", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# h2a_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/H2A_p22_chip/SRR3659059/SRR3659059_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "H2A.z P22", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
# h3_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/H3_p22_chip/SRR3659065/SRR3659065_norm.bw", as="GRanges"), 
#                    genome = "mm10",name = "H3 P22", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 


# make peak tracks
atoh1_zic_track = make_track(atoh1_zic_overlap, "Overlap")
# chd7_zic_track = make_track(chd7_zic_overlap, "Chd7")
# chd4_zic_track = make_track(chd4_zic_overlap, "Chd4")
# ctcf_p4_zic_track = make_track(ctcfP4_zic_overlap, "Ctcf P4")
# ctcf_p22_zic_track = make_track(ctcfP22_zic_overlap, "Ctcf P22")
# ctcf_p56_zic_track = make_track(ctcfP56_zic_overlap, "Ctcf P56")
# rad21_p4_zic_track = make_track(rad21P4_zic_overlap, "Rad21 P4")
# rad21_p56_zic_track = make_track(rad21P56_zic_overlap, "Rad21 P56")
# smc1_zic_track = make_track(smc1_zic_overlap, "Smc1")
# h2a_zic_track = make_track(h2a_zic_overlap, "H2A.z")
# h3_zic_track = make_track(H3_zic_overlap, "H3")
# 


tracks = list(p7_zic_bw1, 
              p7_zic_bw2, 
              p60_zic_bw1, 
              p60_zic_bw2,
              atoh1_bw1,
              atoh1_bw2)



png("../figures/atoh1_overlap_track.png", units = "in", height = 3.5, width = 12, res = 300)
plot_track("Ccnd2", tracks, highlight = c(127112639, 127147731, 127179170), w = 5000, 25000, 50000, ylim = c(0,2)) ## this is the one
dev.off()

png("../figures/atoh1_overlap_track_Chd7.png", units = "in", height = 3.5, width = 12, res = 300)
plot_track("Chd7", tracks, 50000, 50000,highlight = c(8682365, 8748592, 8796800), w =5000, ylim = c(0,2)) ## this is the one
dev.off()

png("../figures/atoh1_overlap_track_Zmiz1.png", units = "in", height = 3.5, width = 12, res = 300)
plot_track("Zmiz1", tracks, 50000, 50000, highlight = c(25419216, 25477487, 25492739, 25509984, 25518084, 25546236), w =5000, ylim = c(0,2)) ## this is the one
dev.off()

png("../figures/atoh1_overlap_track_Dpf3.png", units = "in", height = 3.5, width = 12, res = 300)
plot_track("Dpf3", tracks, 25000, 15000, highlight = c(83208853, 83246662, 83275189, 83300109, 83339703), w =5000, ylim = c(0,2)) ## this is the one
dev.off()


e2 = rasterGrob(readPNG("../figures/atoh1_overlap_track_Dpf3.png"))
e1 = rasterGrob(readPNG("../figures/atoh1_overlap_track_Chd7.png"))
e3 = rasterGrob(readPNG("../figures/atoh1_overlap_track_Zmiz1.png"))
e4 = rasterGrob(readPNG("../figures/atoh1_overlap_track.png"))


plot_grid(e1, e2, e3, e4, ncol=1)
```

# Distinctly Enriched TFs

```{r figure 3, fig.height=12, fig.width=12}
# all peaks
all_fig =  plot_grid(plot_grid( plot_grid(dh1, d1, ncol = 1, labels = c("A", "B"), hjust = 0), d_tf_all , ncol = 2, rel_widths = c(.4, .6) ),
                    plot_grid(plot_grid(atoh_p$plot1, atoh_p$plot2, nrow =2, labels = c("E", "F")), plot_grid(e1, e2, ncol =1, labels = c("G", "H")), rel_widths = c(.3, .7)),
                    ncol = 1,
                    rel_heights = c(.65, .35))



png("../figures/figure2.png", height = 12, width = 12, res = 300, units = "in" )
all_fig
dev.off()

all_fig


```


# Supplemental Figures
```{r supp figure panel A}
homer_tfs = early_homer %>%  
  dplyr::mutate(SYMBOL = str_to_title(SYMBOL)) %>% 
  dplyr::select(SYMBOL) %>% 
  drop_na() %>% 
  distinct() %>% 
  pull(SYMBOL) 


bart_tfs = early_bart %>% 
  dplyr::select(SYMBOL) %>% 
  drop_na() %>% 
  distinct() %>% 
  pull(SYMBOL)


png("../figures/venn_diagram.png")
venn.plot = venn.diagram(
  x = list(bart_tfs, homer_tfs),
  category.names = c("Bart" , "Homer" ),
  filename = NULL,
  fill = c( "#FF6308", "black"),
  label.col = c("black", "blue", "black"),
  fontface ="bold",
  cex = 5,
  cat.cex = 3,
  cat.fontface = "bold",
  cat.dist = c(.005, 0.0005), 
  cat.pos= c(-45, 0),
  rotation.degree = 45, 
  disable.logging  = T
)

grid.draw(venn.plot)
dev.off()

```


```{r supp fig 2, fig.height= 15, fig.width=15}

axis_text_size = 10
sa = rasterGrob(readPNG("../figures/venn_diagram.png"))

sh = plot_bart_homer("early_all", early_all_bart, early_all_homer, axis_text_size, "Early")
si = plot_bart_homer("late_all", late_all_bart, late_all_homer, axis_text_size, "Late")


r1 = rasterGrob(readPNG("../figures/earlyVlate_all_rrho_bart.png"))
r5 = rasterGrob(readPNG("../figures/earlyVlate_all_rrho_homer.png"))


supp = plot_grid(sa,
            plot_grid(plot_grid(sh,si,  nrow = 1, labels = c("B", "C")), plot_grid(r1, r5, nrow = 2, labels = c("D", "E")), nrow =1, rel_widths = c(.7, .3)),
           labels = c("A", ""),
           nrow = 2, 
           rel_heights  = c(.15, .85),
           axis = "l"
           )
 

png("../figures/supp_figure2.png", height = 15, width = 15, res = 300, units = "in" )
supp
dev.off()

supp

```

Supplemental Figure . Mouse in-vivo ChIP overlap enrichement (BART) and motif enrichement (HOMER) of Zic ChIP peaks. A) Overlap in TF ChIP profiles in each algorithm. TF enrichement was calculated by timepoint and gene regulation in the following categories B) Early Activaitng, C) Late Activating D) Early Repressive E) Late Repressive F) Early, E) Late.  *Denotes predicted TFs that are common between the ChIP overlap enrichement and motf enrichment. Color of TFs on y-axis indicate whether that TF was in both datasets (blue) ot not(black).





```{r looped results, fig.height= 15, fig.width=12}

# looped peaks
r2 = rasterGrob(readPNG("../figures/earlyvlate_rrho_bart.png"))
r6 = rasterGrob(readPNG("../figures/earlyvlate_rrho_homer.png"))

sf = plot_bart_homer("early", early_bart, early_homer, axis_text_size, "Early Looped")
sg = plot_bart_homer("late", late_bart, late_homer, axis_text_size, "Late Looped")

looped_fig = plot_grid(
            plot_grid(sf, sg, plot_grid(r2, r6, nrow = 2, labels = c("C", "D")),  labels = c("A", "B"), nrow = 1),
            plot_grid( plot_grid(ch1, c1, ncol = 1, labels = c("C", "D"), hjust = 0), d_tf, rel_widths = c(.4, .6)),
            nrow = 2, 
            rel_heights = c(.6, .4))

png("../figures/figure2_loop.png", height = 15, width = 12, res = 300, units = "in" )
looped_fig
dev.off()


looped_fig
```

```{r  all v looped, fig.height= 8, fig.width=8 }
# all v looped zic peaks
sa4 = plot_distinct(early_allVloop, "early_all", "early", "Distinct \nAll Zic P7", "Distinct \nLooped Zic P7", table_tilte = "Bart\n")
sa5 = plot_distinct(late_allVloop, "late_all", "late", "Distinct \nAll Zic P60", "Distinct \nLooped Zic P60", table_tilte = "Bart\n")
sah4 = plot_distinct(early_allVloop_hom, "early_all", "early", "Distinct \nAll Zic P7", "Distinct \nLooped Zic P7", table_tilte = "Homer\n")
sah5 = plot_distinct(late_allVloop_hom, "late_all", "late", "Distinct \nAll Zic P60", "Distinct \nLooped Zic P60", table_tilte = "Homer\n")

r3 = rasterGrob(readPNG("../figures/early_allVloop_rrho_bart.png"))
r4 = rasterGrob(readPNG("../figures/late_allVloop_rrho_bart.png"))
r7 = rasterGrob(readPNG("../figures/early_allVloop_rrho_homer.png"))
r8 = rasterGrob(readPNG("../figures/late_allVloop_rrho_homer.png"))

supp_allvlooped = plot_grid(
  plot_grid(r3, r4, r7, r8, nrow = 1, labels = LETTERS[1:4]),
  plot_grid(sa4, sah4, labels = c("E", "F"), hjust = -1),
  plot_grid(sa5, sah5, labels = c("G", "H"), hjust = -1), 
  nrow =3)


png("../figures/supp_figure_rrho_allvlooped.png", height = 8, width = 8, res = 300, units = "in" )
supp_allvlooped
dev.off()

supp_allvlooped
```

```{r distinct supp, fig.width=10, fig.height=7}

# Act v Rep
sa1 = plot_distinct(early_actVrep, "early_activating", "early_repressive", "Distinct \nActivating", "Distinct \nRepressive", table_tilte = "Bart\n" )
sa2 = plot_distinct(late_actvrep, "late_activating", "late_repressive", "Distinct \nActivating", "Distinct \nRepressive", table_tilte = "Bart\n" )
sa3 = plot_distinct(actvrep, "activating", "repressive", "Distinct \nActivating", "Distinct \nRepressive", table_tilte = "Bart\n" )
sah1 = plot_distinct(early_actVrep_hom, "early_activating", "early_repressive", "Distinct \nActivating", "Distinct \nRepressive" , table_tilte = "Homer\n")
sah2 = plot_distinct(late_actvrep_hom, "late_activating", "late_repressive", "Distinct \nActivating", "Distinct \nRepressive" , table_tilte = "Homer\n")
sah3 = plot_distinct(actvrep_hom, "activating", "repressive", "Distinct \nActivating", "Distinct \nRepressive" , table_tilte = "Homer\n")



supp_dis = plot_grid(plot_grid(sa1, sa2, sa3, nrow = 1, labels = c("A", "B", "C")), 
          plot_grid(sah1, sah2, sah3, nrow = 1,  labels = c("D", "E", "F")),
          nrow = 2, 
          vjust = 1
)





supp_dis



png("../figures/supp_figure_act_rep.png", height = 8, width = 12, res = 300, units = "in" )
supp_dis
dev.off()

``` 




