---
title: "Figure 2"
output: html_notebook
---

In this document, we reproduce figures from the Zic mansuscript

```{r setup}
library(png)
library(tidyverse)
library(cowplot)
library(gridExtra)
knitr::opts_chunk$set(echo = TRUE)
library("ggpmisc")
library(GenomicRanges)
library(rtracklayer)
library(Gviz) 
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(biomaRt)
library(org.Mm.eg.db)
library(forcats)
library(ggpubr)
library(GenomicInteractions)
library(rstatix)
options(ucscChromosomeNames=FALSE) 
library(ggridges)
```

```{r read data in}
# Mapped peaks to genes via chromatin loops/links
mapped_data = read_tsv("../results/FinalTables/mapped_data.txt")
zic_loop_data = read_tsv("../results/FinalTables/zic_loop_data.txt")
# gene expression data
gene_expr = read_tsv("../results/DiffExp_RNA/normalized_counts.tsv")
# zic peak data (differential)
zic_peak_data <- read_delim("../results/FinalTables/zic_peak_data.txt", delim = "\t", escape_double = FALSE, trim_ws = TRUE)

# Chromatin Links
# > P56 PLAC-seq loops. source: https://www.nature.com/articles/s41586-019-1190-7
loop_data_adult <- read_delim("../sequencing_data/Yamada/combined_MAPS_peaks.txt",
                        delim = "\t", escape_double = FALSE, trim_ws = TRUE, col_names = F) %>% 
                    dplyr::mutate(loop_id = paste0("loop_", 1:n())) 
# > P4 loops derived from Hi-C. source: https://www.nature.com/articles/s41467-021-25846-3
loop_data_young <- read_excel("/media/west-lab-share/Melyssa_Minto/Cerebellum/zic_analysis/sequencing_data/Reddy/GSE164360_Hi-C_Summary.xlsx") %>% 
    dplyr::select(ends_with(c("A", "B"))) %>% 
  set_names(paste0("X",1:6)) %>% 
  dplyr::mutate(loop_id = paste0("loop_", 1:n())) %>% 
  dplyr::mutate(X1 = ifelse(is.na(X1), "chrX", paste0("chr",X1)),
                X4 = ifelse(is.na(X4), "chrX", paste0("chr",X4)))

loop_annotation = read_tsv("../results/FinalTables/loop_data.txt")

```

```{r define functions}
#' Title
#'
#' @param gene - gene to plot
#' @param tracks - GViz tracks to plot
#' @param expand_l - Number of base-pairs to expand to the left
#' @param expand_r - Number of base-pairs to expand to the right
#' @param ylim - y-axis lim
#'
#' @return 
#' @export


plot_track <- function(gene, tracks,  expand_l= 0, expand_r = 0, ylim = c(0,3)){
  # getting gene info
  id = mapIds(org.Mm.eg.db, keys =gene , column ="ENTREZID", keytype="SYMBOL")
  info = biomaRt::select(TxDb.Mmusculus.UCSC.mm10.knownGene, keys = id  , columns=c("TXNAME", "TXSTART", "TXEND", "TXCHROM"), keytype="GENEID")
  start = info$TXSTART[1]
  end = info$TXEND[1]
  chr = info$TXCHROM[1]
  
  # setting up plot inputs
  start = start - expand_l
  end = end + expand_r  
  
  gencode <- UcscTrack(genome = "mm10", chromosome = chr,
                      track = "All GENCODE VM25", from = start, to = end,
                      trackType = "GeneRegionTrack", 
                      rstarts = "exonStarts", rends = "exonEnds", 
                      gene = "name",  symbol = "name2", 
                      transcript = "name", strand = "strand",
                      name = "transcripts", showID=TRUE)
  
  tracks$gencode = gencode
  tracks$axis = genomeAxis

  s = c(2, rep(1, length(tracks)-2), 0.5)
    #plot
    plotTracks(tracks,  
               from=start,
               to=end, 
               chromosome=chr,
               type="hist",
               window = 1000,
               ylim = ylim,
               sizes = s)

  
  
  
}

```

# Supplemental Figure 1
```{r supp fig 1}
# > Anchor Mapping Stats ---------------------------------------------------
# how many anchors are mapped to genes
sa1 = mapped_data %>% 
  dplyr::select(id, gene_name, gene_sig) %>% 
  dplyr::filter(gene_sig != "filtered") %>%
  dplyr::mutate(in_anchor = ifelse(is.na(id), FALSE, TRUE),
                stage = case_when(grepl("adult", id) ~ "adult",
                                  grepl("young", id) ~ "young")) %>% 
  dplyr::select(-id) %>% 
  distinct() %>% 
  group_by(in_anchor, stage) %>% 
  dplyr::count(gene_sig) %>% 
  ggplot(aes(x = gene_sig, y = n, label = n, fill = stage))+
  geom_col() +
  geom_text(vjust = 1, fontface = "bold", color = "black", size = 3)+
  facet_wrap(~in_anchor, labeller = labeller(in_anchor = c("FALSE" = "Not in Anchor", "TRUE" = "With in Anchor"))) +
  labs(y = "# genes", x = "Gene Regulation") +
  theme_bw()+
  theme(plot.title = element_text(face = "bold"),
        legend.position = "bottom")
  
  

# How many zic peaks are not in anchors
sb1 = mapped_data %>% 
  dplyr::select(id, zic_peak, zic_sig) %>% 
  dplyr::filter(zic_sig != "filtered") %>%
  dplyr::filter(!grepl("chr[X|Y]", zic_peak)) %>%
  dplyr::mutate(in_anchor = ifelse(is.na(id), FALSE, TRUE),
                stage = case_when(grepl("adult", id) ~ "adult",
                                  grepl("young", id) ~ "young")) %>% 
  dplyr::select(-id) %>% 
  distinct() %>% 
  group_by(in_anchor) %>% 
  dplyr::count(zic_sig, stage) %>% 
  ggplot(aes(x = zic_sig, y = n, label = n, fill = stage))+
  geom_col() +
  geom_text(vjust = 1, fontface = "bold", color = "black", size = 3)+
  facet_wrap(~in_anchor, labeller = labeller(in_anchor = c("FALSE" = "Not in Anchor", "TRUE" = "With in Anchor"))) +
  labs(y = "# Zic peaks", x = "Zic Regulation") +
  theme_bw()+
  theme(plot.title = element_text(face = "bold"),
        legend.position = "bottom")


# How many DNase peaks are not in anchors
sc1 = mapped_data %>% 
  dplyr::select(id, dnase_peak, dnase_sig) %>%
  dplyr::filter(dnase_sig != "filtered") %>% 
  dplyr::filter(!grepl("chr[X|Y]", dnase_peak)) %>% 
  dplyr::mutate(in_anchor = ifelse(is.na(id), FALSE, TRUE),
                stage = case_when(grepl("adult", id) ~ "adult",
                                  grepl("young", id) ~ "young")) %>% 
  dplyr::select(-id) %>% 
  distinct() %>%
  group_by(in_anchor, stage) %>% 
  dplyr::count(dnase_sig) %>% 
  ggplot(aes(x = dnase_sig, y = n, label = n, fill = stage))+
  geom_col() +
  geom_text(vjust = 2,fontface = "bold", color = "black", size = 3)+
  facet_wrap(~in_anchor, labeller = labeller(in_anchor = c("FALSE" = "Not in Anchor", "TRUE" = "With in Anchor"))) +
  labs(y = "# DHS peaks", x = "DNase Regulation") +
  theme_bw()+
  theme(plot.title = element_text(face = "bold"))


# How many H3K27ac peaks are not in anchors
sd1 = mapped_data %>% 
  dplyr::select(id, k27ac_peak, k27ac_sig) %>%
  dplyr::filter(k27ac_sig != "filtered") %>% 
  dplyr::filter(!grepl("chr[X|Y]", k27ac_peak)) %>% 
  dplyr::mutate(in_anchor = ifelse(is.na(id), FALSE, TRUE),
                stage = case_when(grepl("adult", id) ~ "adult",
                                  grepl("young", id) ~ "young")) %>% 
  dplyr::select(-id) %>% 
  distinct() %>% 
  group_by(in_anchor) %>% 
  dplyr::count(k27ac_sig, stage) %>% 
  ggplot(aes(x = k27ac_sig, y = n, label = n, fill = stage))+
  geom_col() +
  geom_text(vjust = 2, fontface = "bold", color = "black", size = 3)+
  facet_wrap(~in_anchor, labeller = labeller(in_anchor = c("FALSE" = "Not in Anchor", "TRUE" = "With in Anchor"))) +
  labs(y = "# H3K27ac peaks", x = "H3K27ac Regulation") +
  theme_bw() +
  theme(plot.title = element_text(face = "bold"))


#png("../figures/supp_figure2.png", height = 4, width = 7, units = "in", res = 300)
plot_grid(sa1,sb1, nrow = 1, labels = LETTERS[1:4])
#dev.off()

```

Supplemental Figure 2. Mapping statitistics of A) Nearest gene, B) Zic, C) DNase, and D) H3K27ac peaks in H4Kme3 PLAC-seq derived chroamtin loops

# Supplemental Figure 2
```{r supp fig 2, fig.height=7, fig.width=7}
# > How many peaks per genes ----------------------------------------------

n_peaks_to_gene = mapped_data %>% 
  drop_na("id") %>% 
  dplyr::select(zic_peak, gene_name) %>% 
  distinct() %>% 
  dplyr::filter(!is.na(zic_peak)) %>% 
  group_by(gene_name) %>% 
  dplyr::count() %>% 
  ungroup() %>% 
  dplyr::filter(!is.na(gene_name)) %>% 
  left_join(mapped_data %>% dplyr::select(gene_name, gene_sig)) %>% 
  dplyr::mutate(gene_name = fct_reorder(gene_name, n)) %>% 
  dplyr::filter(gene_sig != "filtered") %>% 
  distinct() 


sa2 = n_peaks_to_gene %>% 
  top_n(60, n) %>% 
  ggplot(aes(x = n, y = gene_name, fill =gene_sig))+
  geom_bar(stat = "identity") +
  geom_text(aes(label = n), hjust = 1, color = "white", fontface = "bold") +
  scale_fill_manual( name = "Gene Diff Signal", limits = c("N.S.", "DOWN", "UP"), labels = c("Constituitive", "Down-regulated", "Up-regulated"), values = c("black", "blue", "red"))+
  theme_classic() +
  theme(legend.position = "bottom") +
  labs(y = "")

sb2 = n_peaks_to_gene %>% 
  ggplot(aes(x = n, y= gene_sig))+
  stat_density_ridges()+
  theme_classic() +
  labs( x = "Number of Zic Peaks", y = "Gene Diff Signal")


sc2 =n_peaks_to_gene %>% 
  left_join(mapped_data %>% dplyr::select(starts_with("gene"))) %>% 
  distinct() %>% 
  ggplot(aes(x = n, y = gene_baseMean)) +
  geom_point(alpha = 0.5)+
  geom_smooth(method='lm', se = FALSE)+
  stat_regline_equation(label.y = 100000, aes(label = ..eq.label..)) +
  stat_regline_equation(label.y = 95000, aes(label = ..rr.label..)) +
  theme_classic() +
  labs(x = "Number of Zic Peaks", y = "Mean Normalized \nGene Expression")

sd2 = n_peaks_to_gene %>% 
  left_join(mapped_data %>% dplyr::select(starts_with("gene"))) %>% 
  distinct() %>% 
  ggplot(aes(x = n, y = abs(gene_lfc))) +
  geom_point(alpha = 0.5)+
  geom_smooth(method='lm', se = FALSE)+
  stat_regline_equation(label.y = 20, aes(label = ..eq.label..)) +
  stat_regline_equation(label.y = 19, aes(label = ..rr.label..)) +
  theme_classic() +
  labs(x = "Number of Zic Peaks", y = "Absolute Gene LFC")

#png("../figures/supp_figure3.png", height = 7, width = 10, units = "in", res = 300)
plot_grid(sa2, 
          plot_grid(sb2, sc2,sd2, align = "v", ncol = 1, labels = c("B", "C", "D")),
          ncol = 2,
          nrow = 1,
          labels = c("A", ""),
          rel_widths = c(2, 1))
#dev.off()
```
Supplemental Figure 3. A) Count of Zic peaks mapped to gene. B) Distribution of number of zic peaks by gene regulation.  Scatterplots of C) Mean expression and D) Log2FC of genes against Number of zic peaks show that there is no correlation between number of zic peaks and gene degree of gene regulation. 



# Figure 2
```{r }
gene_loop = loop_annotation %>% 
  dplyr::select(id, proximal= anchor_name.x, distal = anchor_name.y, gene_name) %>% 
  distinct() %>% 
  group_by(gene_name, id) %>% 
  dplyr::mutate(distal = ifelse(proximal %in% distal, NA, distal)) %>%
  ungroup() %>% 
  pivot_longer(cols = c("proximal", "distal"), values_to = "anchor_name", names_to = "anchor_type") %>% 
  drop_na(anchor_name) 


dist_prox = zic_loop_data %>% dplyr::rename(sequence_name = zic_peak) %>% 
  full_join(zic_peak_data,na_matches = "never") %>%
  left_join(gene_loop, by = c("id", "anchor_name"), na_matches = "never") %>% 
  dplyr::select(anchor_name, loop_id = id, sequence_name, zic_PeakID, zic_sig, gene_name, anchor_type) %>% 
  distinct() %>% 
  group_by(loop_id) %>% 
  dplyr::mutate(num_anchors = n_distinct(anchor_name)) %>% 
  dplyr::filter(num_anchors > 1) %>%  
  ungroup() %>% 
  dplyr::select(-loop_id) %>% 
  distinct() %>% 
  group_by(gene_name, anchor_type, zic_sig) %>% 
  summarise(count = n()) %>% 
  ungroup() %>% 
  group_by( gene_name, zic_sig) %>% 
  dplyr::mutate(total = sum(count)) %>% 
  ungroup() %>% 
  group_by(gene_name, zic_sig) %>% 
  dplyr::mutate(ratio = count / total) %>% 
  ungroup() 
  

plot_zic_ratio <- function(genes){
  dist_prox %>%   
  dplyr::filter(gene_name %in% genes) %>% 
  ggplot(aes(x = anchor_type, y = ratio, fill = zic_sig)) +
  geom_col(position = position_dodge2(width = 0.9, preserve = "single")) + 
  theme_bw() +
  scale_fill_manual(values = c( "blue", "black","red"),labels = c("UP" = "P60", "N.S." = "Static","DOWN" = "P7")) +
  facet_wrap(~gene_name) +
  labs(x = "Anchor", y = "# of Peaks")
}


plot_zic_ratio(c("Chd7"))


```

```{r schematic}
sch <-  rasterGrob(readPNG("../figures/schematic.png"), interpolate = T)
```

```{r panel a , fig.height=4, fig.width=3}
# > nearest gene expression -----------------------------------------------

# need fo focus on P60
df = mapped_data %>%
  dplyr::select(gene_name, zic_sig, p60_mean, p7_mean)  %>%
  dplyr::filter(zic_sig != "filtered") %>%
  group_by(zic_sig) %>%
  distinct() %>%
  dplyr::filter(!is.na(zic_sig)) %>%
  pivot_longer(cols = ends_with("mean"), values_to = "means", names_to = "timepoint") %>%
  dplyr::mutate(timepoint = factor(gsub("_mean", "", timepoint)))

stat.test = df %>%
  t_test(means ~ timepoint) %>%
  adjust_pvalue(method = "bonferroni") %>%
  add_significance() %>%
  add_xy_position(fun = "mean_se")
  

plt = df %>% 
  group_by(zic_sig, timepoint) %>% 
  summarise( mean= mean(means, na.rm = T), 
             sd = sd(means, na.rm = T)/sqrt(n()),
             .groups = "drop") %>% 
  ungroup() %>% 
   # plotting
  dplyr::mutate(timepoint = fct_relevel(timepoint, levels = c("p7", "p60") )) %>% 
  ggplot(aes(x = zic_sig, y = mean, ymin = mean - sd, ymax = mean + sd, fill = timepoint, color = timepoint)) +
  geom_errorbar(width = 0.2, position = position_dodge(.8)) +
  scale_fill_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7"))+
  scale_color_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7"))+
  scale_x_discrete(labels=c("DOWN" = "P7 Peak", "N.S." = "Static", "UP" = "P60 Peak"), expand = c(0,0))+
  theme_classic()+
  xlab("Zic Peak Enrichement") +
  ylab("Mean Expresson \nof Mapped Gene") +
  ggsignif::geom_signif(
    tip_length = 0,
    annotations = stat.test$p.adj.signif,
    xmin = c(0.8, 1.8, 2.8),
    xmax = c(1.2,2.2,3.2), 
    y_position = stat.test$y.position,
    color = "black")
  
a1 = plt +
  geom_point(size = 1, position = position_dodge(0.8)) +
  theme(legend.position="bottom",
        legend.title = element_text(size = 8, face = "bold")) +
  labs(fill = "Expression", color = "Expression")
 
  


a2 = plt +
  geom_bar(stat = "identity", position = position_dodge()) +
  theme(legend.position = "bottom",
         legend.text = element_text(size = 8),
         legend.title = element_text(size = 8, face = "bold"),
        legend.key.size = unit(.3, "lines"),
        legend.box.background = element_rect(color = "black")) +
  labs(fill = "Expression", color = "Expression")




a1
a2
```


```{r panel a alt, fig.height=8, fig.width=4}
median.quartile <- function(x){
  out <- quantile(x, probs = c(0.25,0.5,0.75))
  names(out) <- c("ymin","y","ymax")
  return(out)
}

a3 = df %>% 
  #dplyr::filter(zic_sig %in% c("UP", "DOWN")) %>% 
  dplyr::mutate(timepoint = fct_relevel(timepoint, levels = c("p7", "p60") )) %>% 
  ggplot(aes(x = zic_sig, y = means, fill = timepoint, color = timepoint, group = interaction(timepoint, zic_sig))) +
  geom_jitter( position = position_jitterdodge(jitter.width = 0.2, dodge.width = 0.8), size = 1.2, alpha = 0.5)+
  scale_color_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7")) +
  scale_fill_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7")) +
  scale_x_discrete(labels=c("DOWN" = "P7 Peak", "N.S." = "Static", "UP" = "P60 Peak"))+
  theme_classic() 

a4 = df %>% 
  #dplyr::filter(zic_sig %in% c("UP", "DOWN")) %>% 
  dplyr::mutate(timepoint = fct_relevel(timepoint, levels = c("p7", "p60") )) %>% 
  ggplot(aes(x = zic_sig, y = means, fill = timepoint, color = timepoint, group = interaction(timepoint, zic_sig))) +
  stat_summary(fun.data = "mean_se",  fun.args = list(mult = 1),  geom = "pointrange",  position = position_dodge(0.8)) +
  scale_color_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7")) +
  scale_fill_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7")) +
  scale_x_discrete(labels=c("DOWN" = "P7 Peak", "N.S." = "Static", "UP" = "P60 Peak"))+
  theme_classic() +
  ylab("Mean of Means")

a5 = df %>% 
  #dplyr::filter(zic_sig %in% c("UP", "DOWN")) %>% 
  dplyr::mutate(timepoint = fct_relevel(timepoint, levels = c("p7", "p60") )) %>% 
  ggplot(aes(x = zic_sig, y = means, fill = timepoint, color = timepoint, group = interaction(timepoint, zic_sig))) +
  stat_summary(fun.data = median.quartile, geom = "pointrange", position = position_dodge(0.8))+
  scale_color_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7")) +
  scale_fill_manual(values = c( "blue", "red"),labels = c("p60" = "P60", "p7" = "P7")) +
  scale_x_discrete(labels=c("DOWN" = "P7 Peak", "N.S." = "Static", "UP" = "P60 Peak"))+
  theme_classic() +
  ylab("Median of Means")

plot_grid(a3, a4, a5, ncol = 1)
```




```{r panel b, fig.height=4, fig.width=2.5}
# > nearest gene categorical ----------------------------------------------

b = mapped_data %>% 
  drop_na(c("id", "zic_peak", "gene_sig", "zic_sig")) %>% 
  dplyr::select(zic_peak, gene_name, zic_sig, gene_sig) %>% 
  distinct() %>% 
  dplyr::filter( zic_sig != "filtered") %>% 
  dplyr::filter( gene_sig != "filtered") %>% 
  dplyr::count(zic_sig, gene_sig) %>% 
  ggplot(aes(x = zic_sig, y =n,  fill = gene_sig))+
  geom_bar(position = "fill", stat = "identity") +
  geom_text(aes(label=n),position = position_fill(vjust = 0.5), color = "white", fontface = "bold", size = 3) +
  scale_x_discrete(labels=c("DOWN" = "P7 Peak", "N.S." = "Static", "UP" = "P60 Peak"), expand = c(0,0))+
  scale_y_continuous(expand = c(0,0))+
  scale_fill_manual( limits = c("N.S.", "DOWN", "UP"), labels = c("N.S.", "P7", "P60"), values = c("black", "blue", "red"))+
  theme_classic() +
  labs(x = "Zic Peak\n Enrichment", fill = "Gene\nEnrichment", y = "Proportion\n of Genes") +
  theme( legend.position = "bottom",
         legend.text = element_text(size = 6),
         legend.title = element_text(size = 8, face = "bold"),
        legend.key.size = unit(.5, "lines"),
       # legend.box.background = element_rect(color = "black")
        )

b




```



```{r panel c, fig.height=4, fig.width=8}

top_genes = n_peaks_to_gene %>% 
  dplyr::filter(gene_sig != "N.S.") %>% 
  dplyr::arrange(n) %>% 
  group_by(gene_sig) %>% 
  top_n(40, n) %>% 
  pull(gene_name)


c = mapped_data %>% 
  drop_na("id") %>% 
  dplyr::select(zic_peak, gene_name) %>% 
  distinct() %>% 
  dplyr::filter(!is.na(zic_peak)) %>% 
  group_by(gene_name) %>% 
  dplyr::add_count() %>% 
  ungroup() %>% 
  left_join(mapped_data %>% dplyr::select(gene_name, gene_sig, zic_peak, zic_sig) %>%  distinct()) %>% 
  dplyr::filter(gene_name %in% top_genes) %>% 
  dplyr::mutate(gene_name = fct_reorder(gene_name, n))  %>% 
  dplyr::arrange(desc(n)) %>%
  ggplot(aes(y = gene_name, fill =zic_sig))+
  geom_bar(stat = "count", ) +
  #coord_flip()+
  # geom_text(aes(label = n), hjust = 1, color = "white", fontface = "bold") +
  facet_wrap(.~gene_sig, scales = "free", labeller = labeller(gene_sig = c("DOWN" = "Down-regulated", "UP" = "Up-regulated")), ncol = 2)+
  scale_fill_manual( name = "Zic Peak Enrichment", limits = c("DOWN", "N.S.", "UP"), labels = c("P7 Peak", "Static", "P60 Peak"), values = c("blue", "black", "red"))+
  theme_classic() + 
  labs(y = "Gene Symbol", x = "# Zic Peaks Mapped to Gene") +
  theme(axis.text.x = element_text(size = 8, angle = 45, hjust = 1), legend.position = "bottom",
                                                     #legend.justification = c("right", "top"),
                                                     legend.box.just = "right",
                                                     legend.margin = margin(2, 2, 2, 2),
                                                     #legend.box.background = element_rect(color="black", size=1),
                                                     legend.text = element_text(size = 5),
                                                     legend.key.size = unit(.3, "lines"),
                                                     legend.title = element_text(size = 8, face = "bold"))
      


c

```



```{r panel d, fig.width=15, fig.height=6}
## prepare data
 # set scheme
scheme <- getScheme(name=getOption("Gviz.scheme"))
scheme$GdObject$col.axis = "white"
scheme$GdObject$background.title = "black"
scheme$GdObject$fontcolor = "black"
scheme$GdObject$fontsize = 18
scheme$GdObject$rotation.title = 90
scheme$GeneRegionTrack$transcriptAnnotation = "symbol"
scheme$GeneRegionTrack$col = "black"
scheme$GeneRegionTrack$fill = "black"
scheme$GeneRegionTrack$collapseTranscripts = FALSE
scheme$DataTrack$fill.histogram = "darkblue"
scheme$DataTrack$col.histogram = "darkblue"
scheme$GdObject$col.border.title = "transparent"

addScheme(scheme, "myScheme")
options(Gviz.scheme="myScheme")

# mm10 genome info
genomeAxis <- GenomeAxisTrack(name="MyAxis") 
customFromTxDb <- GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene) 

# knownGenes <- UcscTrack(genome = "mm10", chromosome = "chrX", 
#                         track = "knownGene", from = from, to = to,
#                         trackType = "GeneRegionTrack", 
#                         rstarts = "exonStarts", rends = "exonEnds", 
#                         gene = "name", symbol = "name", 
#                         transcript = "name", strand = "strand", 
#                         fill = "#8282d2", name = "UCSC Genes")

all_loops = bind_rows(loop_data_adult, loop_data_young)

interaction_track <- InteractionTrack(GenomicInteractions(GRanges(seqnames = all_loops$X1, ranges = IRanges(all_loops$X2, all_loops$X3)), 
                                                          GRanges(seqnames = all_loops$X4, ranges = IRanges(all_loops$X5, all_loops$X6))), 
                                      name = "Chromatin Loops")

# Zic ChIP, H3K27ac ChIP, DHS-seq, RNA-seq
# > Source: Frank et al. 2015 https://www.nature.com/articles/nn.3995
# > data was reprocessed to mm10 and reanalyzed 
p7_zic_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557091/SRR1557091_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nP7") 
p60_zic_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557093/SRR1557093_norm.bw", as="GRanges"), 
                    genome = "mm10", name = "Zic \nP60") 
# p7_dnase_bw <-  DataTrack(import.bw("../sequencing_data/preprocessed_data/DNase/SRR1557041/SRR1557041.bw", as="GRanges"),
#                            genome = "mm10",name = "DNase \nP7") 
# p60_dnase_bw <-  DataTrack(import.bw("../sequencing_data/preprocessed_data/DNase/SRR1557047/SRR1557047.bw", as="GRanges"), 
#                             genome = "mm10",name = "DNase \nP60") 
# p7_k27ac_bw <-  DataTrack(import.bw("../sequencing_data/preprocessed_data/H3K27ac/SRR3993161/SRR3993161_norm.bw", as="GRanges"), 
#                            genome = "mm10",name = "H3K27ac \nP7") 
# p60_k27ac_bw <-  DataTrack(import.bw("../sequencing_data/preprocessed_data/H3K27ac/SRR3993163/SRR3993163_norm.bw", as="GRanges"), 
#                             genome = "mm10",name = "H3K27ac \nP60") 
p7_rna_bw <-  DataTrack(import.bw("../sequencing_data/preprocessed_data/RNA/p7_b1/p7_b1_norm.bw", as="GRanges"), 
                           genome = "mm10",name = "RNA \nP7") 
p60_rna_bw <-  DataTrack(import.bw("../sequencing_data/preprocessed_data/RNA/p60_b1/p60_b1_norm.bw", as="GRanges"), 
                            genome = "mm10",name = "RNA \nP60") 



## plot
  tracks <- list(
              interaction_track,
               p7_zic_bw,
               p60_zic_bw, 
               # p7_dnase_bw,
               # p60_dnase_bw,
               # p7_k27ac_bw,
               # p60_k27ac_bw,
               p7_rna_bw,
               p60_rna_bw)
  
  

# plot
# plot_track("Grin2c",tracks,  expand_l = 100000, expand_r = 1000)
# 
# plot_track("Wnt7b", tracks,  expand_l = 100000, expand_r = 10000)
# 
# plot_track("Sez6l", tracks,  expand_l = 10000, expand_r = 10000)
# 
# plot_track("Chd7", tracks,  expand_l = 10000, expand_r = 250000)


png("../figures/fig3_d.png", units = "in", height = 6, width = 15, res = 300)
plot_track("Nr4a3", tracks,  expand_l = 600000, expand_r = 5000, ylim = c(0,2))
dev.off()

d = rasterGrob(readPNG("../figures/fig3_d.png"))

```
![](../figures/fig3_d.png)
```{r figure 2, fig.height=15, fig.width=15}


fig3 = plot_grid( plot_grid( sch,  plot_grid(b, a1, ncol = 1, labels = c("B", "C")), c, rel_widths = c(.5, .15, .35), nrow = 1,labels = c("A", "", "D")),
           d,
           ncol = 1,
           labels = c( "","E" ),
           rel_heights = c(.50, .50)
           )

png("../figures/figure3.png", height = 12, width = 12, res = 300, units = "in")
fig3
dev.off()

fig3
```
Figure 2. Zic peaks were mapped to genes via H4Kme3 mediated chromatin loops. A) Overall number of genes mapped to Zic peaks that are enriched at P7, P60, and Statically bound. B) Mean expression of genes at P7 and P60 mapped to P7 and P60 enriched Zic peaks. C) Number of Zic peaks mapped to genes that are differentially expressed between P7 and P60. D) Example tracks of H4Kme3 loops, Zic, DHS, H3k27ac, and Gene expression at P7 and P60
