---
title: "Figure_invitro"
output: html_notebook
---



```{r setup, include=FALSE}
library(tidyverse)
library(ggrepel)
library(cowplot)
library(clusterProfiler)
library(ggtext)
library(org.Mm.eg.db)
library(DESeq2)
library(grid)
library(png)
library(GenomicInteractions)
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(rtracklayer)
library(Gviz)

 # set scheme
scheme <- getScheme(name=getOption("Gviz.scheme"))
scheme$GdObject$col.axis = "white"
scheme$GdObject$background.title = "black"
scheme$GdObject$fontcolor = "black"
scheme$GdObject$rotation.title = 90
scheme$GeneRegionTrack$transcriptAnnotation = "symbol"
scheme$GeneRegionTrack$col = "black"
scheme$GeneRegionTrack$fill = "black"
scheme$GeneRegionTrack$collapseTranscripts = FALSE
scheme$DataTrack$fill.histogram = "darkblue"
scheme$DataTrack$col.histogram = "darkblue"
scheme$GdObject$col.border.title = "transparent"
scheme$HighlightTrack$col = "grey"
scheme$HighlightTrack$fill = "#e3e8ff"
addScheme(scheme, "myScheme")
options(Gviz.scheme="myScheme")
options(ucscChromosomeNames=FALSE)
```

# Direct target assignments focusing on in vitro data.

Idea is to get the direct targets assignments using only in vitro data.

# Read datasets

```{r read data in, echo = FALSE,warning = FALSE,message = FALSE, error = FALSE}
invitro_data = read_tsv("../results/FinalTables/invitro_table.txt")
go_results =  load("../results/invitro/figure_invitro/go_cat.RData")
zic_cuntnrun = read_rds("../results/invitro/diffbind_cutnrun_zic/res_7v3_ashr.Rds")
zic_cuntnrun7v1 = read_rds("../results/invitro/diffbind_cutnrun_zic/res_7v1_ashr.Rds")
zic_cuntnrun7v5 = read_rds("../results/invitro/diffbind_cutnrun_zic/res_7v5_ashr.Rds")
zic_cuntnrun3v1 = read_rds("../results/invitro/diffbind_cutnrun_zic/res_3v1_ashr.Rds")
zic_cuntnrun5v3 = read_rds("../results/invitro/diffbind_cutnrun_zic/res_5v3_ashr.Rds")
zic_cuntnrun5v1 = read_rds("../results/invitro/diffbind_cutnrun_zic/res_5v1_ashr.Rds")


zic_cutnrun_anno_up = read_rds("../results/invitro/diffbind_cutnrun_zic/peaks_7v3_up_annotation.Rds")
zic_cutnrun_anno_down = read_rds("../results/invitro/diffbind_cutnrun_zic/peaks_7v3_down_annotation.Rds")
zic_cutnrun_anno_static = read_rds("../results/invitro/diffbind_cutnrun_zic/peaks_7v3_static_annotation.Rds")

div3v7_expr = read_rds("../results/invitro/diffexpr/res_d7_d3_ashr.Rds")
zic1kd_expr = read_rds("../results/invitro/diffexpr/res_zic1_ashr.Rds")
zic2kd_expr = read_rds("../results/invitro/diffexpr/res_zic2_ashr.Rds")

deseq_obj_cutnrun_union <- readRDS("../results/invitro/diffbind_cutnrun_zic/deseq_obj_chip_cutnrun.Rds")

# Chromatin Links
# > P56 PLAC-seq loops. source: https://www.nature.com/articles/s41586-019-1190-7
loop_data_adult <- read_delim("../sequencing_data/Yamada/combined_MAPS_peaks.txt",
                        delim = "\t", escape_double = FALSE, trim_ws = TRUE, col_names = F) %>% 
                    dplyr::mutate(loop_id = paste0("loop_", 1:n())) 
# > P4 loops derived from Hi-C. source: https://www.nature.com/articles/s41467-021-25846-3
loop_data_young <- readxl::read_excel("../sequencing_data/Reddy/GSE164360_Hi-C_Summary.xlsx") %>% 
    dplyr::select(ends_with(c("A", "B"))) %>% 
  set_names(paste0("X",1:6)) %>% 
  dplyr::mutate(loop_id = paste0("loop_", 1:n())) %>% 
  dplyr::mutate(X1 = ifelse(is.na(X1), "chrX", paste0("chr",X1)),
                X4 = ifelse(is.na(X4), "chrX", paste0("chr",X4)))

# TF enrichment results
dev_bart = read_tsv("../results/invitro/bart_results/dev_anchors/bart_results.txt")
zd_bart  = read_tsv("../results/invitro/bart_results/zd_anchors/bart_results.txt")
zdd_bart = read_tsv("../results/invitro/bart_results/zdd_anchors/bart_results.txt")

devVzdd_bart = read_tsv("../results/invitro/beds/rrho/devVzdd_anchors.txt")
devVzd_bart = read_tsv("../results/invitro/beds/rrho/devVzd_anchors.txt")
zdVzdd_bart = read_tsv("../results/invitro/beds/rrho/zdVzdd_anchors.txt")
devVzdd_homer = read_tsv("../results/invitro/beds/rrho/devVzdd_anchors_homer.txt")
devVzd_homer = read_tsv("../results/invitro/beds/rrho/devVzd_anchors_homer.txt")
zdVzdd_homer = read_tsv("../results/invitro/beds/rrho/zdVzdd_anchors_homer.txt")


```




```{r define functions}
get_overlaps_by_element <- function(peaks_a, peaks_b) {
                              intersection1 <- sum(IRanges::overlapsAny(peaks_a,peaks_b))
                              setdiff1 <- length(peaks_a) - intersection1
                              intersection2 <- sum(IRanges::overlapsAny(peaks_b,peaks_a))
                              setdiff2 <-length(peaks_b) - intersection2
                              
                              return(data.frame("intersection1" = intersection1,
                                                "intersection2" = intersection2,
                                                "setdiff1" = setdiff1,
                                                "setdiff2" = setdiff2))}

plot_volcano<- function(data, height, up, down, comp){
  data = 
    data %>% 
    as_tibble() %>% 
    dplyr::mutate(sig = case_when(log2FoldChange > 0 & padj <= 0.05 ~ "UP",
                         log2FoldChange < 0 & padj <= 0.05 ~ "DOWN",
                         is.na(padj) ~ "filtered",
                         TRUE ~ "N.S.")) 
  
  tab = table(data$sig)
  cat(tab)

  data %>% 
    dplyr::filter(sig != "filtered") %>% 
    ggplot(aes(x = log2FoldChange, y = -log10(padj), color = sig)) +
    geom_point() +
    scale_color_manual(limits = c("DOWN", "N.S.", "UP"), values = c("blue", "black", "red"), labels = c("Early", "Static", "Late")) +
    labs(y = "log10(adjusted p-value)", x = paste0(comp, " Log2FC")) +
    geom_text(x = up, y = height, label = paste0("Up-regulated\n n = ",tab["UP"]), color = "red")+
    geom_text(x = down, y = height, label = paste0("Down-regulated\n n = ",tab["DOWN"] ), color = "blue")+
    theme_classic() +
    theme(legend.position = "none")
}

plot_ma <- function(data, x, up, down, comp, color_up = "red", color_down = "blue"){
  plt_dat = data %>% 
  as_tibble() %>% 
  dplyr::mutate(sig = case_when(log2FoldChange > 0 & padj <= 0.05 ~ "UP",
                         log2FoldChange < 0 & padj <= 0.05 ~ "DOWN",
                         is.na(padj) ~ "filtered",
                         TRUE ~ "N.S.")) 

  tab = plt_dat %>% 
    group_by(sig) %>% 
    dplyr::count() %>% 
    ungroup()
    
  num_up = tab$n[tab$sig == "UP"]
  num_down = tab$n[tab$sig == "DOWN"]
  
  tab
  
  total = tab %>% 
    dplyr::filter(sig != "filtered") %>% 
    summarize(total = sum(n))
    
  plt_dat %>% 
    dplyr::filter(sig != "filtered") %>% 
    ggplot(aes(y = log2FoldChange, x = log10(baseMean), color = sig)) +
    geom_point() +
    scale_color_manual(limits = c("DOWN", "N.S.", "UP"), values = c(color_down, "black", color_up)) +
    labs(x = "log10(Avg Zic CutnRun Signal)", y = paste0(comp, " Log2FC")) +
    geom_text(x = x, y = up, label = paste0("Increased Zic Binding\n n = ",num_up), color = color_up)+
    geom_text(x = x, y = down, label = paste0("Decreased Zic Binding\n n = ",num_down ), color = color_down)+
    theme_classic() +
    theme(legend.position = "none")

}

plot_track <- function(gene, tracks, highlight=NULL, w = 1000, expand_l= 0, expand_r = 0, ylim = c(0,3), highlight_gene=F, g_exp_l=1000, g_exp_r=0){
  # getting gene info
  id = mapIds(org.Mm.eg.db, keys =gene , column ="ENTREZID", keytype="SYMBOL")
  info = biomaRt::select(TxDb.Mmusculus.UCSC.mm10.knownGene, keys = id  , columns=c("TXNAME", "TXSTART", "TXEND", "TXCHROM"), keytype="GENEID")
  start_g = info$TXSTART[1]
  end_g = info$TXEND[1]
  chr = info$TXCHROM[1]
  
  # setting up plot inputs
  start = start_g - expand_l
  end = end_g + expand_r  
  cat(chr, start, end)
  # getting gene track
  gencode <- UcscTrack(genome = "mm10", chromosome = chr,
                      track = "All GENCODE VM25", from = start, to = end,
                      trackType = "GeneRegionTrack", 
                      rstarts = "exonStarts", rends = "exonEnds", 
                      gene = "name",  symbol = "name2", 
                      transcript = "name", strand = "strand",
                      name = "transcripts", showID=TRUE)
 
  
    # > highlight track
  if(is.null(highlight)){
    # create trackset for plotting
    ts = tracks 
    ts[ length(tracks) +1 ] = gencode
    ts[length(tracks) + 2] = genomeAxis
  }else{
    # creating trackset for plotting with highlighting
    ht1 <- HighlightTrack(trackList = tracks, start = highlight, width = w, chromosome = chr)
    ts = list(ht1, gencode, genomeAxis)
  }
  
  # highlight gene
  if(highlight_gene){
    ht1 <- HighlightTrack(trackList = gencode, start = start_g-g_exp_l, width = (end_g - start_g)+ g_exp_r, chromosome = chr)
  
    ts = tracks 
    ts[ length(tracks) +1 ] = ht1
    ts[length(tracks) + 2] = genomeAxis
  }
  

    #plot
    # > create a vector of track sizes
    s = c(rep(1, length(tracks) ), 0.5, 1)
   Gviz::plotTracks(trackList=ts,  
               from=start,
               to=end, 
               chromosome=chr,
               type="hist",
               window = 1000,
               ylim = ylim,
               sizes = s)
  
  
}

distinct_lolipop  <- function(data, cond1, cond2, gene_expr_level) {
  # plot set up
   
  # extract data
  d1 = data %>%  
    # determine whether each TF is enriched in one set v another set
    dplyr::mutate(enrich_sig = ifelse(enriched_in == cond1, get(cond1), get(cond2))) %>% 
    # filter for significant TFs in set1
    dplyr::filter(enriched_in %in% c(cond1)) %>% 
    # remove significance column
    dplyr::select(-all_of(cond1)) %>% 
    # cleaning up TF names and creating label for plotting
    dplyr::mutate(TF = gsub("-", "", TF)) %>% 
    dplyr::mutate(label = ifelse(str_to_lower(TF) != str_to_lower(SYMBOL), paste0(TF, "(", SYMBOL, ")"), TF),
                  label = fct_reorder(label, -enrich_sig)) 
  
  d2 =  data %>% 
    # determine whether each TF is enriched in one set v another set
    dplyr::mutate(enrich_sig = ifelse(enriched_in == cond1, get(cond1), get(cond2))) %>% 
    # filter for significant TFs in set2
    dplyr::filter(enriched_in %in% c(cond2)) %>% 
     # remove significance column
    dplyr::select( -all_of(cond2)) %>% 
    # cleaning up TF names and creating label for plotting
    dplyr::mutate(TF = gsub("-", "", TF)) %>% 
    dplyr::mutate(label = ifelse(str_to_lower(TF) != str_to_lower(SYMBOL), paste0(TF, "(", SYMBOL, ")"), TF),
                  label = fct_reorder(label, -enrich_sig))
  
  # plotting data
    bind_rows(
      list( Late = d2  %>%
              # filter out ot time specific gene expression
              dplyr::filter(p60_mean > gene_expr_level) %>% 
              # filter for time specific gene regulation
              dplyr::filter(gene_sig %in% c("UP")) %>% 
              # sort TF by log fold change
              dplyr::mutate(label = fct_reorder(label, log2FoldChange)) %>% 
              # renaming gene means for plotting
              dplyr::rename(plt_mean = p60_mean),
            Early = d1 %>% 
              # filter out ot time specific gene expression
              dplyr::filter(p7_mean > gene_expr_level) %>% 
               # filter for time specific gene regulation
              dplyr::filter(gene_sig %in% c( "DOWN")) %>% 
              # sort TF by log fold change
              dplyr::mutate(label = fct_reorder(label, -log2FoldChange))%>% 
              # renaming gene means for plotting
              dplyr::rename(plt_mean = p7_mean)),
      .id = "id") %>% 
    # plot  
    ggplot(aes(y = log2FoldChange , x = label, color = enrich_sig, size = plt_mean )) +
    geom_segment( aes(x=label, xend=label, y=0, yend=log2FoldChange, size = 1), color = "grey", show.legend = FALSE) +
    geom_hline(yintercept = 0, color = "black", size = .5, linetype = "dashed") +
    geom_point() +
    theme_classic() +
    scale_x_discrete(labels = function(x) str_wrap(x, width = 5))+
    scale_color_viridis(limits = c(0, 0.05))+
    facet_grid(id~., scales = "free", space = "free_y", labeller = labeller(facet_labels)) +
    coord_flip() +
    theme(legend.position = "bottom", 
          legend.text = element_text(angle = 45, hjust = 1),
          strip.text.x = element_markdown() , 
          axis.text.y = element_text(size = 8),
          panel.spacing=unit(0,"pt"), 
          plot.title = element_markdown()) +
    labs(x = "Enriched TFs" , y = "LFC", color = "TF Enrichment", size = "expression")   

}

```
# Plots
In  Culture Gene Expression Stats

```{r expression volcano plots, fig.height=4, fig.width=12}
volc_a = plot_volcano(div3v7_expr, 39, 4, -3, "DIV7/DIV3")
volc_b = plot_volcano(zic1kd_expr, 21, 4.5, -2, "Zic1 KD v WT (DVI7)")
volc_c = plot_volcano(zic2kd_expr, 25, 2.5, -2, "Zic2 KD v WT (DVI7)")

plot_grid(volc_a, volc_b, volc_c, nrow = 1)
```
```{r expression rrho}
rrho_zic1 = rasterGrob(readPNG("../results/invitro/rrho/zic1_dev_RRHO_neg.png"))
rrho_zic2 = rasterGrob(readPNG("../results/invitro/rrho/zic2_dev_RRHO_neg.png"))

plot_grid(rrho_zic1, rrho_zic2)

```
```{r supp fig, fig.height=6, fig.width=10}
supp = plot_grid(volc_a, volc_b, volc_c,rrho_zic1,rrho_zic2,  nrow = 2, labels = LETTERS[1:5])

# png("../figures/supp_figure_vitro.png", height = 6, width = 10, res = 300, units = "in")
# supp
# dev.off()

supp
```

Zic Cut n run stats
```{r cutnrun MA}


ma  = plot_ma(zic_cuntnrun, 2.25, 1.5, -1.85, "DIV7/DIV3")
ma7v1  = plot_ma(zic_cuntnrun7v1, 2.5, 1.5, -1.85, "DIV7/DIV1" )
ma7v5  = plot_ma(zic_cuntnrun7v5, 2.5, .5, -.5, "DIV7/DIV5")
ma3v1  = plot_ma(zic_cuntnrun3v1, 2.5, 2, -1.1, "DIV3/DIV1")
ma5v1  = plot_ma(zic_cuntnrun5v1, 2.5, 1.5, -1.85, "DIV5/DIV1")
ma5v3  = plot_ma(zic_cuntnrun5v3, 2.5, .5, -.7, "DIV5/DIV3")


ma
```


```{r suopp fig ma, fig.height=10, fig.width=12}

supp2 = plot_grid(ma7v1, ma7v5,ma3v1, ma5v1, ma5v3, nrow = 3, labels = LETTERS[1:5] )

# png("../figures/supp_figure_cutrun.png", height = 10, width = 12, res = 300, units = "in")
# supp2
# dev.off()

supp2
```

# comparing zic cut n run in vivo to zic chip
```{r cutnrun peak size}

size = bind_rows(list(Early = zic_cutnrun_anno_down@anno %>% as_tibble(),
               Late = zic_cutnrun_anno_up@anno %>% as_tibble(),
               Static = zic_cutnrun_anno_static@anno %>% as_tibble()),
          .id = "id") %>% 
  dplyr::mutate(size = end - start) %>% 
   ggplot(aes(x = size) )+ 
  geom_histogram(fill = "white", color = "black", bins = 50) +
  theme_classic() +
  xlim(0, 4000)+
  labs(x = "Size of Zic Peaks", y = "Num. Peaks")

size
```

```{r cutnrun loci}

loci = bind_rows(list(DIV7= zic_cutnrun_anno_up@annoStat, DIV3 = zic_cutnrun_anno_down@annoStat, Static = zic_cutnrun_anno_static@annoStat),.id = "id") %>% 
  dplyr::mutate(Feature2 = case_when( grepl("Promoter", Feature) ~ "Promoter Proximal",
                                      Feature %in% c("Downstream (<=300)", "Distal Intergenic") ~ "Distal",
                                      TRUE ~ "Gene Body"),
                Feature2 = factor(Feature2, levels = c("Promoter Proximal", "Gene Body", "Distal"))) %>% 
  ggplot(aes(x = id, y = Frequency, fill = Feature2))+
  geom_col() +
  labs(y = "% Peaks", x = "Differential Zic Region", fill = "Region") +
  theme_classic() +
  scale_x_discrete(limits = c("DIV3", "Static", "DIV7")) +
  scale_fill_manual(values = c("#bad2e1","grey", "#b15928"), labels = c("Promoter\nProximal", "Gene\nBody", "Distal"))+
  theme( panel.grid = element_blank(), legend.position = "right", legend.key.size = unit(.2, 'cm')) +
  scale_y_continuous(expand = c(0,0), labels = scales::percent_format(scale = 1))
loci

```

```{r PCA}
## Prefiltering
keep <- rowSums(counts(deseq_obj_cutnrun_union)) >= 20
deseq_obj_cutnrun_union <- deseq_obj_cutnrun_union[keep,]

## VST
deseq_cutnrun_vst <- vst(deseq_obj_cutnrun_union[,1:12],blind = TRUE)
deseq_chip_vst <- vst(deseq_obj_cutnrun_union[,13:16],blind = TRUE)
deseq_vst <- vst(deseq_obj_cutnrun_union,blind = TRUE)
```


```{r PCA,fig.height=12, fig.width=4}
## plot PCA
pca_vivo = plotPCA(object = deseq_cutnrun_vst,intgroup=c("time")) + 
  theme_classic() +
  theme(legend.title = element_blank())+
  scale_color_manual(limits = c("D1Z", "D3Z", "D5Z", "D7Z"), labels = c("DIV1", "DIV3", "DIV5", "DIV7"), values = c("#fbd2ff", "#ff80ee", "#cb80ff", "#008fff"))

pca_vitro = plotPCA(object = deseq_chip_vst,intgroup=c("time")) + 
  theme_classic() +
  theme(legend.title = element_blank())+
  scale_color_manual(limits = c( "P7", "P60"), labels = c("P7", "P60"), values = c( "red", "blue"))

pca_all = plotPCA(object = deseq_vst,intgroup=c("time")) + 
  theme_classic() +
  theme(legend.title = element_blank())+
  scale_color_manual(limits = c("D1Z", "D3Z", "D5Z", "D7Z", "P7", "P60"), labels = c("DIV1", "DIV3", "DIV5", "DIV7", "P7", "P60"), values = c("#fbd2ff", "#ff80ee", "#cb80ff", "#008fff", "red", "blue"))

plot_grid(pca_vivo,pca_vitro, pca_all, ncol=1)

```
```{r peak overlap}

# get in vitro peaks
chip_macs_peaks_dir <- "../results/invivo/DiffExp_ZicChIP/"
chip_macs_peaks_files <- list.files(chip_macs_peaks_dir,pattern = "P60")
chip_macs_peaks_full <- paste0(chip_macs_peaks_dir,chip_macs_peaks_files)
chip_macs_peaks_names <- c("P7","NS","P60")

chip_macs_granges <- lapply(chip_macs_peaks_full, import.bedGraph)

names(chip_macs_granges) <- chip_macs_peaks_names

# get in vitro peaks
peaks_DIV3 <- rtracklayer::import("../results/invitro/diffbind_cutnrun_zic/res_7v3_down.bed")
peaks_DIV7 <- rtracklayer::import("../results/invitro/diffbind_cutnrun_zic/res_7v3_up.bed")

diff_peaks_overlaps_by_element <- rbind(get_overlaps_by_element(peaks_DIV3, chip_macs_granges$P7),
get_overlaps_by_element(peaks_DIV3, chip_macs_granges$P60),
get_overlaps_by_element(peaks_DIV7, chip_macs_granges$P7),
get_overlaps_by_element(peaks_DIV7, chip_macs_granges$P60))


diff_peaks_overlaps_by_element$set1 <- c("DIV3" , "DIV3", "DIV7", "DIV7")
diff_peaks_overlaps_by_element$set2 <- c("Early", "Late", "Early", "Late")

ov = diff_peaks_overlaps_by_element %>%
  dplyr::select(-setdiff2, -intersection2) %>%
  pivot_longer(where(is.numeric)) %>%
  group_by(set1,set2) %>%
  mutate(percentage = 100 * value / sum(value)) %>%
  ungroup() %>%
  mutate(name = fct_relevel(name,"setdiff1","intersection1"),
         name = fct_recode(name, "overlap" = "intersection1", "no overlap" = "setdiff1"),
         set2 = fct_relevel(set2,"Early","Late")) %>%
  filter(name == "overlap") %>%
  ggplot(aes(y = percentage, x = set2, fill = set2)) +
    geom_col(width = 0.5) +
    ylab(NULL) +
    ylab("Percentage") +
    xlab(NULL) +
    scale_fill_manual(values = c("Early"="red","Late"="blue")) +
    theme_bw() +
    theme(legend.title = element_blank(),
        strip.background = element_rect(fill = "white"),
        strip.text = element_text(color = "black", size = 12)) +
    facet_grid(~set1) +
    ylim(c(0,100)) +
    geom_text(aes(label = paste0(round(percentage),"%")),nudge_y = 2)

ov
```
```{r supp chip-cutnrun, fig.width=8, fig.height=6}
colrs = c("#fbd2ff", "#ff80ee", "#cb80ff", "#008fff", "red", "blue")
names(colrs) = c("DIV1", "DIV3", "DIV5", "DIV7", "P7", "P60")

ma7v3  = plot_ma(zic_cuntnrun, 2.25, 1.5, -1.85, "DIV7/DIV3",   as.character(colrs["DIV7"]),as.character(colrs["DIV3"]))
ma7v1  = plot_ma(zic_cuntnrun7v1, 2.5, 1.5, -1.85, "DIV7/DIV1", as.character(colrs["DIV7"]),as.character(colrs["DIV1"]))
ma7v5  = plot_ma(zic_cuntnrun7v5, 2.5, .5, -.5, "DIV7/DIV5",    as.character(colrs["DIV7"]),as.character(colrs["DIV5"]))
ma3v1  = plot_ma(zic_cuntnrun3v1, 2.5, 2, -1.1, "DIV3/DIV1",    as.character(colrs["DIV3"]),as.character(colrs["DIV1"]))
ma5v1  = plot_ma(zic_cuntnrun5v1, 2.5, 1.5, -1.85, "DIV5/DIV1", as.character(colrs["DIV5"]),as.character(colrs["DIV1"]))
ma5v3  = plot_ma(zic_cuntnrun5v3, 2.5, .5, -.7, "DIV5/DIV3",    as.character(colrs["DIV5"]),as.character(colrs["DIV3"]))
```


```{r supp chip-cutnrun, fig.width=15, fig.height=15}
supp_cc = plot_grid(ma7v1, ma7v3, ma7v5,ma3v1, ma5v1, ma5v3,pca_vivo,pca_vitro, pca_all, ov, ncol = 3, labels = LETTERS[1:10] )

png("../figures/supp_figure_cutrun.png", height = 15, width = 15, res = 300, units = "in")
supp_cc
dev.off()

supp_cc
```


```{r supp chip-cutnrun, fig.width=8, fig.height=6}
plot_grid(pca_vivo,pca_vitro, pca_all, ov, ncol=2, labels = LETTERS[1:4])

```


## Explore ZDD genes
Zic1 log2FC vs number of in vitro peaks
```{r num peak to genes, fig.height=8, fig.width=12}
a = invitro_data %>% 
  dplyr::select( gene_names, dev_log2foldchange, dev_padj,  zic_peak, num_peaks) %>% 
  distinct() %>% 
  dplyr::select(-zic_peak) %>% 
  dplyr::mutate(dev_sig = ifelse(abs(dev_log2foldchange) >=2 & dev_padj < 0.05, "sig", "ns")) %>% 
  distinct() %>%
  ggplot(aes( x= num_peaks, y = dev_log2foldchange, label = gene_names, color = dev_sig)) +
  geom_point(alpha = 0.5, size = 1) +
  scale_color_manual(values = c("grey40", "black"))+
  geom_text_repel() +
  theme_bw() +
  xlab("Number of mapped invitro Zic peaks") +
  ylab("GE log2(DIV7 / DIV3)")


b = invitro_data %>% 
  dplyr::select(id, gene_names, log2foldchange, padj, zic_peak, num_peaks) %>% 
  distinct() %>% 
  dplyr::select(-zic_peak) %>% 
  dplyr::mutate(kd_sig = ifelse(abs(log2foldchange) >=2 & padj < 0.05, "sig", "ns")) %>% 
  distinct() %>% 
  ggplot(aes( x= num_peaks, y = log2foldchange, label = gene_names, color = kd_sig)) +
  geom_point(alpha = 0.3, size = 1) +
  scale_color_manual(values = c("grey40", "black"))+
  geom_text_repel()+
  theme_bw() +
  facet_wrap(~id)+
  xlab("Number of mapped invitro Zic peaks") +
  ylab("GE log2(KD / Ctrl DIV7)")

plot_grid(a,b, ncol = 1)

```



Zic1 KD/Ctrl vs D7/D3 


```{r zic dependent genes, fig.height=8, fig.width=12}
c = invitro_data %>% 
  dplyr::select(zic_peak, id, dev_log2foldchange,log2foldchange,gene_names, bothKD, cat, num_peaks ) %>% 
  distinct() %>% 
  dplyr::select(-zic_peak) %>% 
  distinct() %>% 
  ggplot(aes( x = dev_log2foldchange, y = log2foldchange, label = gene_names, color = bothKD, size = num_peaks)) +
  geom_point(alpha = 0.5) +
  geom_text_repel(size = 5, min.segment.length = unit(5, "cm"), max.overlaps = 20)+
  facet_grid(id~cat) +
  theme_bw() +
  scale_color_manual(values = c("#143589", "#E1BE6A", "#40B0A6"), limits =  c("Zic1 KD", "Zic2 KD", "Zic1 & Zic2 KD"))+
  theme(legend.position = "bottom",
        strip.background = element_rect(fill = "white"),
        strip.text = element_text(color = "black", size = 12))+
  labs(x = "GE log2(DIV7 / DIV3)", y = "GE log2(Zic1/2 KD / Ctrl)", color = "KD", size = "no. mapped peaks" )

c

```

```{r}
invitro_data %>% 
  dplyr::select(gene_names,cat) %>% 
  distinct() %>% 
  group_by(cat) %>% 
  dplyr::count()
```




```{r ZDD peak to genes, fig.width= 6, fig.height=8}


d1 = dotplot(zdd_t_bp) + 
  theme(axis.text.y = element_text(size = 11), axis.text.x = element_text(size =11)) + 
  scale_y_discrete(labels = function(x) str_wrap(x, width = 25)) +
  scale_x_discrete( labels = str_wrap(c("Fail to Up-regulate", "Fail to Down-regulate"), width = 15)) +
  scale_color_viridis_c(option = "magma")

d2 = cnetplot(zdd_t_bp, showCategory = 5, color_gene = "green", cex_label_gene = 1.25, cex_label_category = 1.6, layout = "fr") + 
  scale_fill_grey(limits = c("Up", "Down"), labels = c("Fail to Up-regulate", "Fail to Down-regulate")) 

d2

plot_grid(d1, d2, ncol = 1)


 
  
```



Adding in vitro n peaks

```{r, fig.height=12, fig.width=8}
plt_dat = invitro_data %>% 
  # select data
  dplyr::select(ends_with(c("hange", "adj")),  zic_peak, gene_names, bothKD, cat , num_peaks, dev_sig, zic_sig) %>% 
  distinct() %>% 
  # color genes by affected KD 
  dplyr::mutate(gene_labels = case_when(bothKD == "Zic1 KD" ~ paste0('<span style="color:#143589">', gene_names,'</span>' ),
                                       bothKD == "Zic2 KD" ~ paste0('<span style="color:#E1BE6A">', gene_names,'</span>' ),
                                       bothKD == "Zic1 & Zic2 KD" ~ paste0('<span style="color: #40B0A6">', gene_names,'</span>' ))) %>% 

  dplyr::mutate(gene_names = fct_reorder(gene_names, num_peaks)) %>%
  dplyr::mutate(gene_labels = fct_reorder(gene_labels, num_peaks)) %>%
  # filter for plot
  dplyr::select(gene_names,gene_labels, zic_peak, num_peaks, ends_with("sig"), cat) %>% 
  distinct() %>% 
  #dplyr::filter(num_peaks > 20) %>%
  dplyr::filter(dev_sig != "N.S.") %>%
  drop_na(zic_peak) 

e_leg = get_legend(plt_dat %>% 
  dplyr::filter(cat == "Developmental") %>% 
  # plot
  ggplot(aes(y = gene_labels, fill =zic_sig))+
  geom_bar(stat = "count") +
  facet_grid(dev_sig~cat, scales = "free", space = "free")+
  scale_fill_manual(values = c("Up"="red","Down"="blue","Static" ="black")) +
  theme_classic() + 
  labs(y = "Gene Symbol", x = "# Zic Peaks Mapped to Gene", fill = "DIV7/DIV3") +
  theme(axis.text.y = element_markdown(),
        axis.text.x = element_markdown(size = 8), 
        legend.position = "bottom",
         #legend.justification = c("right", "top"),
         legend.box.just = "right",
         legend.margin = margin(2, 2, 2, 2),
         #legend.box.background = element_rect(color="black", size=1),
         legend.text = element_text(size = 11),
         legend.key.size = unit(.5, "lines"),
         legend.title = element_text(size = 11, face = "bold")))

e1 = plt_dat %>% 
  dplyr::filter(cat == "Developmental") %>% 
  # plot
  ggplot(aes(y = gene_labels, fill =zic_sig))+
  geom_bar(stat = "count") +
  facet_grid(dev_sig~cat, scales = "free", space = "free")+
  scale_fill_manual(values = c("Up"="red","Down"="blue","Static" ="black")) +
  theme_classic() + 
  labs(y = "Gene Symbol", x = "# Zic Peaks Mapped to Gene", fill = "DIV7/DIV3") +
  theme(axis.text.y = element_markdown(),
        axis.text.x = element_markdown(size = 8), 
        legend.position = "none")

e2 = plt_dat %>% 
  dplyr::filter(cat == "Zic Dependent Developmental") %>% 
  # plot
  ggplot(aes(y = gene_labels, fill =zic_sig))+
  geom_bar(stat = "count") +
  facet_grid(dev_sig~cat, scales = "free", space = "free", labeller = labeller(dev_sig = c("Down" = "Fail to Down-regulate", "Up"="Fail to Up-regulate")))+
  scale_fill_manual(values = c("Up"="red","Down"="blue","Static" ="black"), labels = c("DIV7", "DIV3", "Static"), limits = c("Up", "Down", "Static")) +
  theme_classic() + 
  labs(y = "", x = "# Zic Peaks Mapped to Gene", fill = "DIV7/DIV3") +
  theme(axis.text.y = element_markdown(size = 10),
        axis.text.x = element_markdown(size = 8), 
        strip.text = element_text(size = 12))
      
      

e = plot_grid(plot_grid(e1, e2),
              e_leg,ncol = 1,
              rel_heights = c(.95,.05))
  
e
e2
```

```{r}
plt_dat %>% 
  dplyr::select(gene_names, cat) %>% 
  distinct() %>% 
  group_by(cat) %>% 
  dplyr::count()
```

```{r example track of ZDD genes}

# formatting data
genomeAxis <- GenomeAxisTrack(name="MyAxis") 
customFromTxDb <- GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene)

# make bigwig tracks
D3_zic_bw1 <- DataTrack(import.bw("../sequencing_data/invitro_data/cutnrun_zic/bws/R110-D3Z_S10_L001_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nDIV3", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
D3_zic_bw2 <- DataTrack(import.bw("../sequencing_data/invitro_data/cutnrun_zic/bws/R122-D3Z_S22_L001_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nDIV3", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
D3_zic_bw3 <- DataTrack(import.bw("../sequencing_data/invitro_data/cutnrun_zic/bws/R130-D3Z_S30_L001_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nDIV3", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 

D7_zic_bw1 <- DataTrack(import.bw("../sequencing_data/invitro_data/cutnrun_zic/bws/R112-D7Z_S12_L001_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nDIV7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
D7_zic_bw2 <- DataTrack(import.bw("../sequencing_data/invitro_data/cutnrun_zic/bws/R124-D7Z_S24_L001_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nDIV7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
D7_zic_bw3 <- DataTrack(import.bw("../sequencing_data/invitro_data/cutnrun_zic/bws/R132-D7Z_S32_L001_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic \nDIV7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 

# interaction track
all_loops = bind_rows(loop_data_adult, loop_data_young)

interaction_track <- InteractionTrack(GenomicInteractions(GRanges(seqnames = all_loops$X1, ranges = IRanges(all_loops$X2, all_loops$X3)), 
                                                          GRanges(seqnames = all_loops$X4, ranges = IRanges(all_loops$X5, all_loops$X6))), 
                                      name = "Chromatin Loops")

## plot
tracks <- list(
            interaction_track,
            D3_zic_bw1,
            D3_zic_bw2,
            D3_zic_bw3,
            D7_zic_bw1,
            D7_zic_bw2,
            D7_zic_bw3)
```

```{r, fig.width=12, fig.height=6}
png("../figures/fig4_example_Nhlh1.png", units = "in", height = 4, width = 12, res = 300)
plot_track("Nhlh1", tracks,  expand_l = 25000, expand_r = 400000, ylim = c(0,2),highlight_gene = T, g_exp_l = 18000, g_exp_r = 25000) #
dev.off()

png("../figures/fig4_example_Ets2.png", units = "in", height = 4, width = 12, res = 300)
plot_track("Ets2", tracks,  expand_l = 75000, expand_r = 265000, ylim = c(0,2),highlight_gene = T, g_exp_l = 18000, g_exp_r = 25000) #
dev.off()

```
```{r}
track_zdd1 = rasterGrob(readPNG("../figures/fig4_example_Nhlh1.png"))
track_zdd2 = rasterGrob(readPNG("../figures/fig4_example_Ets2.png"))

```
![](../figures/fig4_example_Nhlh1.png)

![](../figures/fig4_example_Ets2.png)
```{r TF enrichment, fig.height=10, fig.width=12}
tf_b_1 = dev_bart %>% 
  dplyr::mutate(TF = fct_reorder(TF, -log10(dev_anchors))) %>% 
  dplyr::mutate(gene_sig = case_when(log2FoldChange > 1 & padj < 0.05 ~ "7",
                log2FoldChange < -1 & padj < 0.05 ~ "3",
                is.na(padj) ~ "filtered",
                TRUE ~ "Static")) %>% 
    dplyr::filter(bart_sig %in% "Enriched") %>% 
    dplyr::filter(gene_sig != "filtered") %>% 
    dplyr::filter(!grepl("zic", TF, ignore.case = T)) %>% 
    dplyr::select(TF, gene_sig, ends_with("anchors")) %>% 
  ggplot(aes(y = TF, x = dev_anchors, fill = dev_anchors)) +
    geom_col() +
    theme_classic() +
    facet_grid(gene_sig~., scales = "free", space = "free") +
    labs( x = "adjusted p-value", y = "Developmental TFs")


tf_b_2 = zd_bart %>% 
  dplyr::mutate(TF = fct_reorder(TF, -log10(zd_anchors))) %>% 
  dplyr::mutate(gene_sig = case_when(log2FoldChange > 1 & padj < 0.05 ~ "7",
                log2FoldChange < -1 & padj < 0.05 ~ "3",
                is.na(padj) ~ "filtered",
                TRUE ~ "Static")) %>% 
    dplyr::filter(bart_sig %in% "Enriched") %>% 
    dplyr::filter(gene_sig != "filtered") %>% 
    dplyr::filter(!grepl("zic", TF, ignore.case = T)) %>% 
  dplyr::select(TF, gene_sig, ends_with("anchors")) %>% 
  distinct() %>% 
  ggplot(aes(y = TF, x = zd_anchors, fill = zd_anchors)) +
    geom_col() +
    theme_classic() +
    facet_grid(gene_sig~., scales = "free", space = "free") +
    labs( x = "adjusted p-value", y = "Zic Dependent TFs")

tf_b_3 = zdd_bart %>% 
  dplyr::mutate(TF = fct_reorder(TF, -log10(zdd_anchors))) %>% 
  dplyr::mutate(gene_sig = case_when(log2FoldChange > 1 & padj < 0.05 ~ "7",
                log2FoldChange < -1 & padj < 0.05 ~ "3",
                is.na(padj) ~ "filtered",
                TRUE ~ "Static")) %>% 
    dplyr::filter(bart_sig %in% "Enriched") %>% 
    dplyr::filter(gene_sig != "filtered") %>% 
    dplyr::filter(!grepl("zic", TF, ignore.case = T)) %>% 
  dplyr::select(TF, gene_sig, ends_with("anchors")) %>% 
  distinct() %>% 
  ggplot(aes(y = TF, x = zdd_anchors, fill = zdd_anchors)) +
    geom_col() +
    theme_classic() +
    facet_grid(gene_sig~., scales = "free", space = "free") +
    labs( x = "adjusted p-value", y = "Zic Dependent Developmental TFs")

plot_grid(tf_b_1, tf_b_2, tf_b_3, nrow =1)
```
```{r}
rrho_b1 = rasterGrob(readPNG("../figures/devVzdd_anchors.png"))
rrho_b2 = rasterGrob(readPNG("../figures/zdvzdd_anchors.png"))
rrho_b3 = rasterGrob(readPNG("../figures/devVzd_anchors.png"))
```

![](../figures/devVzdd_anchors.png)

![](../figures/zdVzdd_anchors.png)

![](../figures/devVzd_anchors.png)

```{r TF distinct enrichment, fig.height= 8, fig.width=15}
b1 = devVzdd_bart %>% 
  dplyr::mutate(label = ifelse(enriched_in != "Similar Enrichment",TF, "" )) %>% 
  ggplot(aes(x = -log10(dev_anchors), y = -log10(zdd_anchors), color = enriched_in, label = label)) +
  geom_point() +
  ggrepel::geom_text_repel() +
  theme_classic() +
  labs(x = "Developmental Gene Anchors -log10(adj p-val)", y = "ZDD Gene Anchors -log10(adj p-val)")  +
    scale_color_manual(name = "", limits = c("dev_anchors", "zdd_anchors", "Similar Enrichment"), labels = c("Dev.","ZDD", "Similar \nEnrichment"), values = c("dodgerblue4", "darkorange4", "grey") ) +
    theme(legend.position = "bottom") +
    guides(colour = guide_legend(override.aes = list(size=3)))


b2 = zdVzdd_bart %>% 
  dplyr::mutate(label = ifelse(enriched_in != "Similar Enrichment",TF, "" )) %>% 
  ggplot(aes(x = -log10(zd_anchors), y = -log10(zdd_anchors), color = enriched_in, label = label)) +
  geom_point() +
  ggrepel::geom_text_repel() +
  theme_classic() +
  labs(x = "Zic Dependent Gene Anchors -log10(adj p-val)", y = "ZDD Gene Anchors -log10(adj p-val)")  +
    scale_color_manual(name = "", limits = c("zd_anchors", "zdd_anchors", "Similar Enrichment"), labels = c("ZD","ZDD", "Similar \nEnrichment"), values = c("dodgerblue4", "darkorange4", "grey") ) +
    theme(legend.position = "bottom") +
    guides(colour = guide_legend(override.aes = list(size=3)))

b3 = devVzd_bart %>% 
  dplyr::mutate(label = ifelse(enriched_in != "Similar Enrichment",TF, "" )) %>% 
  ggplot(aes(x = -log10(dev_anchors), y = -log10(zd_anchors), color = enriched_in, label = label)) +
  geom_point() +
  ggrepel::geom_text_repel() +
  theme_classic() +
  labs(x = "Developmental Gene Anchors -log10(adj p-val)", y = "ZD Gene Anchors -log10(adj p-val)")  +
    scale_color_manual(name = "", limits = c("dev_anchors", "zd_anchors", "Similar Enrichment"), labels = c("Dev.","ZD", "Similar \nEnrichment"), values = c("dodgerblue4", "darkorange4", "grey") ) +
    theme(legend.position = "bottom") +
    guides(colour = guide_legend(override.aes = list(size=3)))


h1 = devVzdd_homer %>% 
  dplyr::mutate(label = ifelse(enriched_in != "Similar Enrichment",TF, "" )) %>% 
  ggplot(aes(x = -log10(dev_anchors), y = -log10(zdd_anchors), color = enriched_in, label = label)) +
  geom_point() +
  ggrepel::geom_text_repel() +
  theme_classic() +
  labs(x = "Developmental Gene Anchors -log10(adj p-val)", y = "ZDD Gene Anchors -log10(adj p-val)")  +
    scale_color_manual(name = "", limits = c("dev_anchors", "zdd_anchors", "Similar Enrichment"), labels = c("Dev.","ZDD", "Similar \nEnrichment"), values = c("dodgerblue4", "darkorange4", "grey") ) +
    theme(legend.position = "bottom") +
    guides(colour = guide_legend(override.aes = list(size=3)))


h2 = zdVzdd_homer %>% 
  dplyr::mutate(label = ifelse(enriched_in != "Similar Enrichment",TF, "" )) %>% 
  ggplot(aes(x = -log10(zd_anchors), y = -log10(zdd_anchors), color = enriched_in, label = label)) +
  geom_point() +
  ggrepel::geom_text_repel() +
  theme_classic() +
  labs(x = "Zic Dependent Gene Anchors -log10(adj p-val)", y = "ZDD Gene Anchors -log10(adj p-val)")  +
    scale_color_manual(name = "", limits = c("zd_anchors", "zdd_anchors", "Similar Enrichment"), labels = c("ZD","ZDD", "Similar \nEnrichment"), values = c("dodgerblue4", "darkorange4", "grey") ) +
    theme(legend.position = "bottom") +
    guides(colour = guide_legend(override.aes = list(size=3)))

h3 = devVzd_homer %>% 
  dplyr::mutate(label = ifelse(enriched_in != "Similar Enrichment",TF, "" )) %>% 
  ggplot(aes(x = -log10(dev_anchors), y = -log10(zd_anchors), color = enriched_in, label = label)) +
  geom_point() +
  ggrepel::geom_text_repel() +
  theme_classic() +
  labs(x = "Developmental Gene Anchors -log10(adj p-val)", y = "ZD Gene Anchors -log10(adj p-val)")  +
    scale_color_manual(name = "", limits = c("dev_anchors", "zd_anchors", "Similar Enrichment"), labels = c("Dev.","ZD", "Similar \nEnrichment"), values = c("dodgerblue4", "darkorange4", "grey") ) +
    theme(legend.position = "bottom") +
    guides(colour = guide_legend(override.aes = list(size=3)))


plot_grid(b1, b2, b3, h1, h2, h3, nrow = 2)
```


```{r TF enrichment, fig.height= 15, fig.width=15}
supp_gene_dis = plot_grid( plot_grid(tf_b_1, tf_b_2, tf_b_3, nrow =1, labels = c("A", "B", "C")),
           plot_grid(rrho_b3, rrho_b1,rrho_b2,  nrow =1, labels = c("D", "E", "F")),
           plot_grid(b3, b1,b2, nrow = 1, labels = c("G", "H", "I")),
           ncol = 1,
           rel_heights = c(.6, .15, .25)
)

png("../figures/supp_figure_distinct.png", height = 15, width = 15, res = 300, units = "in")
supp_gene_dis
dev.off()

supp_gene_dis
```

```{r fig4, fig.width=15, fig.height=15}
tracks_fig = plot_grid(track_zdd1, track_zdd2, ncol = 1, labels = c("F", "G"), vjust = 1)
go_fig = plot_grid(d1, d2, labels = c("H", "I"), ncol =1, rel_heights = c(.6, .4))

fig4 = plot_grid(plot_grid(plot_grid( c, plot_grid(ma, size, loci, nrow =1, labels = c("B", "C", "D"), rel_widths = c(.5, .2, .3)), ncol =1,  rel_heights = c(.7, .3), labels = c("A", "")),
           e2,
           labels = c("", "E"),
           rel_widths = c(.7, .3),
           nrow = 1),
 plot_grid(tracks_fig, go_fig, nrow =1, rel_widths = c(.65, .35)),
 rel_heights = c(.55, .45),
 ncol = 1)


png("../figures/figure4.png", height = 15, width = 15, res = 300, units = "in")
fig4
dev.off()
fig4
```
