---
title: "Zic Peaks"
output: html_notebook
---



```{r}
knitr::opts_chunk$set(echo = TRUE)
library("ggpmisc")
library(GenomicRanges)
library(rtracklayer)
library(Gviz) 
library(TxDb.Mmusculus.UCSC.mm10.knownGene)
library(biomaRt)
library(org.Mm.eg.db)
library(forcats)
options(ucscChromosomeNames=FALSE) 

```


```{r}
P60vP7_zic = read_tsv("../results/DiffExp_ZicChIP/ZicChIPDA_data.tsv")
```




```{r}
#' Title
#'
#' @param gene - gene to plot
#' @param tracks - GViz tracks to plot
#' @param expand_l - Number of base-pairs to expand to the left
#' @param expand_r - Number of base-pairs to expand to the right
#' @param ylim - y-axis lim
#'
#' @return 
#' @export
plot_track_coords <- function(tracks, chr, start, end , ylim = c(0,3)){
 
  gencode <- UcscTrack(genome = "mm10", chromosome = chr,
                      track = "All GENCODE VM25", from = start, to = end,
                      trackType = "GeneRegionTrack", 
                      rstarts = "exonStarts", rends = "exonEnds", 
                      gene = "name",  symbol = "name2", 
                      transcript = "name", strand = "strand",
                      name = "transcripts", showID=TRUE)
  
  tracks$gencode = gencode

  
    #plot
    plotTracks(tracks,  
               from=start,
               to=end, 
               chromosome=chr,
               type="hist",
               window = 1000,
               ylim = ylim)

  
  
  
}


plot_track <- function(gene, tracks,  expand_l= 0, expand_r = 0, ylim = c(0,3)){
  # getting gene info
  id = mapIds(org.Mm.eg.db, keys =gene , column ="ENTREZID", keytype="SYMBOL")
  info = biomaRt::select(TxDb.Mmusculus.UCSC.mm10.knownGene, keys = id  , columns=c("TXNAME", "TXSTART", "TXEND", "TXCHROM"), keytype="GENEID")
  start = info$TXSTART[1]
  end = info$TXEND[1]
  chr = info$TXCHROM[1]
  
  # setting up plot inputs
  start = start - expand_l
  end = end + expand_r  
  
  gencode <- UcscTrack(genome = "mm10", chromosome = chr,
                      track = "All GENCODE VM25", from = start, to = end,
                      trackType = "GeneRegionTrack", 
                      rstarts = "exonStarts", rends = "exonEnds", 
                      gene = "name",  symbol = "name2", 
                      transcript = "name", strand = "strand",
                      name = "transcripts", showID=TRUE)
  
  tracks$gencode = gencode

  
    #plot
    plotTracks(tracks,  
               from=start,
               to=end, 
               chromosome=chr,
               type="hist",
               window = 1000,
               ylim = ylim)

  
  
  
}
```

```{r}
# set scheme
scheme <- getScheme(name=getOption("Gviz.scheme"))
scheme$GdObject$col.axis = "white"
scheme$GdObject$background.title = "black"
scheme$GdObject$fontcolor = "black"
scheme$GdObject$rotation.title = 90
scheme$GeneRegionTrack$transcriptAnnotation = "symbol"
scheme$GeneRegionTrack$col = "black"
scheme$GeneRegionTrack$fill = "black"
scheme$GeneRegionTrack$collapseTranscripts = FALSE
scheme$DataTrack$fill.histogram = "darkblue"
scheme$DataTrack$col.histogram = "darkblue"
scheme$GdObject$col.border.title = "transparent"

addScheme(scheme, "myScheme")
options(Gviz.scheme="myScheme")
```

```{r}
# formatting data
genomeAxis <- GenomeAxisTrack(name="MyAxis") 
customFromTxDb <- GeneRegionTrack(TxDb.Mmusculus.UCSC.mm10.knownGene)

p7_zic_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557091/SRR1557091_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic P7", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 
p60_zic_bw <- DataTrack(import.bw("../sequencing_data/preprocessed_data/zic_chip/SRR1557093/SRR1557093_norm.bw", as="GRanges"), 
                   genome = "mm10",name = "Zic P60", col.title = "white", col.border.title = "black", fill.histogram = "darkblue", col.histogram = "darkblue") 


zic_merged = AnnotationTrack(import.bed("../results/DiffExp_ZicChIP/zic.bed"),
                genome = "mm10", name = "Merged", col.title = "white", col.border.title = "black",fill = "purple")

zic_up = AnnotationTrack(import.bed("../results/DiffExp_ZicChIP/P60vP7_UP.bed"),
                genome = "mm10", name = "P60", col.title = "white", col.border.title = "black",fill = "red")

zic_down = AnnotationTrack(import.bed("../results/DiffExp_ZicChIP/P60vP7_DOWN.bed"),
                genome = "mm10", name = "P7", col.title = "white", col.border.title = "black",fill = "blue")

zic_static = AnnotationTrack(import.bed("../results/DiffExp_ZicChIP/P60vP7_NS.bed"),
                genome = "mm10", name = "Static", col.title = "white", col.border.title = "black",fill =  "black")


tracks = list(zic_merged,
              zic_up, 
              zic_down, 
              zic_static,
              p7_zic_bw, 
              p60_zic_bw)

tmp = P60vP7_zic %>% 
  dplyr::arrange(desc(log2FoldChange)) %>% 
  dplyr::select(Chr, Start, End) %>% 
  slice(1)
```





```{r, fig.height=3, fig.width=12}
plot_track("Bcl6",  tracks , 50000, 50000, ylim = c(0,1)) 
```
```{r, fig.height=3, fig.width=12}
plot_track("Fzd7",  tracks , 50000, 50000, ylim = c(0,1)) 
```


```{r, fig.height=3, fig.width=12}
plot_track("Car10",  tracks , 50000, 50000, ylim = c(0,1)) 
```
```{r, fig.height=3, fig.width=12}
plot_track("Klf13",  tracks , 50000, 50000, ylim = c(0,1)) 
```
```{r, fig.height=3, fig.width=12}
plot_track("Coro2b",  tracks , 50000, 50000, ylim = c(0,1)) 
```

```{r, fig.height=3, fig.width=12}
plot_track("Kcnc1",  tracks , 50000, 50000, ylim = c(0,1)) 
```
```{r, fig.height=3, fig.width=12}
plot_track("Negr1",  tracks , 50000, 50000, ylim = c(0,1)) 
```

```{r, fig.height=3, fig.width=12}
plot_track("Grin2c",  tracks , 50000, 50000, ylim = c(0,1)) 

```
```{r, fig.height=3, fig.width=12}
plot_track("Gabra6",  tracks , 50000, 50000, ylim = c(0,1)) 

```